// react-datepicker@4.25.0/dist/es/index.mjs
import h, { createRef as Et } from "react";
import T from "../../../../../classnames@2.5.1/node_modules/classnames/index.mjs";
import We from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/isDate/index.mjs";
import me from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/isValid/index.mjs";
import Qe from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/format/index.mjs";
import Ie from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/addMinutes/index.mjs";
import Nt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/addHours/index.mjs";
import we from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/addDays/index.mjs";
import Oe from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/addWeeks/index.mjs";
import q from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/addMonths/index.mjs";
import xt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/addQuarters/index.mjs";
import ie from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/addYears/index.mjs";
import Yt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/subDays/index.mjs";
import He from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/subWeeks/index.mjs";
import re from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/subMonths/index.mjs";
import Tt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/subQuarters/index.mjs";
import pe from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/subYears/index.mjs";
import je from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/getSeconds/index.mjs";
import A from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/getMinutes/index.mjs";
import W from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/getHours/index.mjs";
import It from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/getDay/index.mjs";
import Ve from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/getDate/index.mjs";
import Ot from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/getISOWeek/index.mjs";
import I from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/getMonth/index.mjs";
import Z from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/getQuarter/index.mjs";
import w from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/getYear/index.mjs";
import Re from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/getTime/index.mjs";
import Rt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/setSeconds/index.mjs";
import ue from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/setMinutes/index.mjs";
import de from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/setHours/index.mjs";
import R from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/setMonth/index.mjs";
import X from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/setQuarter/index.mjs";
import B from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/setYear/index.mjs";
import qe from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/min/index.mjs";
import Ue from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/max/index.mjs";
import fe from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/differenceInCalendarDays/index.mjs";
import ge from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/differenceInCalendarMonths/index.mjs";
import ye from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/differenceInCalendarYears/index.mjs";
import te from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/startOfDay/index.mjs";
import Lt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/startOfWeek/index.mjs";
import ht from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/startOfMonth/index.mjs";
import Ft from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/startOfQuarter/index.mjs";
import mt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/startOfYear/index.mjs";
import Le from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/endOfDay/index.mjs";
import At from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/endOfMonth/index.mjs";
import Wt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/endOfYear/index.mjs";
import Kt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/isEqual/index.mjs";
import Bt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/isSameDay/index.mjs";
import Qt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/isSameMonth/index.mjs";
import Ht from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/isSameYear/index.mjs";
import jt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/isSameQuarter/index.mjs";
import U from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/isAfter/index.mjs";
import H from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/isBefore/index.mjs";
import ce from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/isWithinInterval/index.mjs";
import ft from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/toDate/index.mjs";
import _e from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/parse/index.mjs";
import Vt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/parseISO/index.mjs";
import be from "../../../../../react-onclickoutside@6.13.1_react-dom@18.2.0_react@18.2.0/node_modules/react-onclickoutside/dist/react-onclickoutside.es.mjs";
import qt from "react-dom";
import { Popper as Ut } from "../../../../../react-popper@2.3.0_@popperjs_core@2.11.8_react-dom@18.2.0_react@18.2.0/node_modules/react-popper/lib/esm/Popper.mjs";
import { Manager as $t } from "../../../../../react-popper@2.3.0_@popperjs_core@2.11.8_react-dom@18.2.0_react@18.2.0/node_modules/react-popper/lib/esm/Manager.mjs";
import { Reference as zt } from "../../../../../react-popper@2.3.0_@popperjs_core@2.11.8_react-dom@18.2.0_react@18.2.0/node_modules/react-popper/lib/esm/Reference.mjs";
import Gt from "../../../../../date-fns@2.30.0/node_modules/date-fns/esm/set/index.mjs";
function $e(u, a) {
  var i = Object.keys(u);
  if (Object.getOwnPropertySymbols) {
    var t = Object.getOwnPropertySymbols(u);
    a && (t = t.filter(function(e) {
      return Object.getOwnPropertyDescriptor(u, e).enumerable;
    })), i.push.apply(i, t);
  }
  return i;
}
function ve(u) {
  for (var a = 1; a < arguments.length; a++) {
    var i = arguments[a] != null ? arguments[a] : {};
    a % 2 ? $e(Object(i), !0).forEach(function(t) {
      p(u, t, i[t]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(u, Object.getOwnPropertyDescriptors(i)) : $e(Object(i)).forEach(function(t) {
      Object.defineProperty(u, t, Object.getOwnPropertyDescriptor(i, t));
    });
  }
  return u;
}
function Fe(u) {
  return Fe = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(a) {
    return typeof a;
  } : function(a) {
    return a && typeof Symbol == "function" && a.constructor === Symbol && a !== Symbol.prototype ? "symbol" : typeof a;
  }, Fe(u);
}
function E(u, a) {
  if (!(u instanceof a))
    throw new TypeError("Cannot call a class as a function");
}
function ze(u, a) {
  for (var i = 0; i < a.length; i++) {
    var t = a[i];
    t.enumerable = t.enumerable || !1, t.configurable = !0, "value" in t && (t.writable = !0), Object.defineProperty(u, gt(t.key), t);
  }
}
function N(u, a, i) {
  return a && ze(u.prototype, a), i && ze(u, i), Object.defineProperty(u, "prototype", { writable: !1 }), u;
}
function p(u, a, i) {
  return (a = gt(a)) in u ? Object.defineProperty(u, a, { value: i, enumerable: !0, configurable: !0, writable: !0 }) : u[a] = i, u;
}
function ne() {
  return ne = Object.assign ? Object.assign.bind() : function(u) {
    for (var a = 1; a < arguments.length; a++) {
      var i = arguments[a];
      for (var t in i)
        Object.prototype.hasOwnProperty.call(i, t) && (u[t] = i[t]);
    }
    return u;
  }, ne.apply(this, arguments);
}
function x(u, a) {
  if (typeof a != "function" && a !== null)
    throw new TypeError("Super expression must either be null or a function");
  u.prototype = Object.create(a && a.prototype, { constructor: { value: u, writable: !0, configurable: !0 } }), Object.defineProperty(u, "prototype", { writable: !1 }), a && Ae(u, a);
}
function De(u) {
  return De = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(a) {
    return a.__proto__ || Object.getPrototypeOf(a);
  }, De(u);
}
function Ae(u, a) {
  return Ae = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(i, t) {
    return i.__proto__ = t, i;
  }, Ae(u, a);
}
function l(u) {
  if (u === void 0)
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  return u;
}
function Y(u) {
  var a = function() {
    if (typeof Reflect > "u" || !Reflect.construct || Reflect.construct.sham)
      return !1;
    if (typeof Proxy == "function")
      return !0;
    try {
      return Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function() {
      })), !0;
    } catch {
      return !1;
    }
  }();
  return function() {
    var i, t = De(u);
    if (a) {
      var e = De(this).constructor;
      i = Reflect.construct(t, arguments, e);
    } else
      i = t.apply(this, arguments);
    return function(n, s) {
      if (s && (typeof s == "object" || typeof s == "function"))
        return s;
      if (s !== void 0)
        throw new TypeError("Derived constructors may only return object or undefined");
      return l(n);
    }(this, i);
  };
}
function le(u) {
  return function(a) {
    if (Array.isArray(a))
      return Me(a);
  }(u) || function(a) {
    if (typeof Symbol < "u" && a[Symbol.iterator] != null || a["@@iterator"] != null)
      return Array.from(a);
  }(u) || function(a, i) {
    if (a) {
      if (typeof a == "string")
        return Me(a, i);
      var t = Object.prototype.toString.call(a).slice(8, -1);
      if (t === "Object" && a.constructor && (t = a.constructor.name), t === "Map" || t === "Set")
        return Array.from(a);
      if (t === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t))
        return Me(a, i);
    }
  }(u) || function() {
    throw new TypeError(`Invalid attempt to spread non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`);
  }();
}
function Me(u, a) {
  (a == null || a > u.length) && (a = u.length);
  for (var i = 0, t = new Array(a); i < a; i++)
    t[i] = u[i];
  return t;
}
function gt(u) {
  var a = function(i, t) {
    if (typeof i != "object" || i === null)
      return i;
    var e = i[Symbol.toPrimitive];
    if (e !== void 0) {
      var n = e.call(i, t || "default");
      if (typeof n != "object")
        return n;
      throw new TypeError("@@toPrimitive must return a primitive value.");
    }
    return (t === "string" ? String : Number)(i);
  }(u, "string");
  return typeof a == "symbol" ? a : String(a);
}
var Ge = function(u, a) {
  switch (u) {
    case "P":
      return a.date({ width: "short" });
    case "PP":
      return a.date({ width: "medium" });
    case "PPP":
      return a.date({ width: "long" });
    default:
      return a.date({ width: "full" });
  }
}, Je = function(u, a) {
  switch (u) {
    case "p":
      return a.time({ width: "short" });
    case "pp":
      return a.time({ width: "medium" });
    case "ppp":
      return a.time({ width: "long" });
    default:
      return a.time({ width: "full" });
  }
}, Jt = { p: Je, P: function(u, a) {
  var i, t = u.match(/(P+)(p+)?/) || [], e = t[1], n = t[2];
  if (!n)
    return Ge(u, a);
  switch (e) {
    case "P":
      i = a.dateTime({ width: "short" });
      break;
    case "PP":
      i = a.dateTime({ width: "medium" });
      break;
    case "PPP":
      i = a.dateTime({ width: "long" });
      break;
    default:
      i = a.dateTime({ width: "full" });
  }
  return i.replace("{{date}}", Ge(e, a)).replace("{{time}}", Je(n, a));
} }, se = 12, Xt = /P+p+|P+|p+|''|'(''|[^'])+('|$)|./g;
function b(u) {
  var a = u ? typeof u == "string" || u instanceof String ? Vt(u) : ft(u) : /* @__PURE__ */ new Date();
  return Q(a) ? a : null;
}
function Q(u, a) {
  return a = a || /* @__PURE__ */ new Date("1/1/1000"), me(u) && !H(u, a);
}
function _(u, a, i) {
  if (i === "en")
    return Qe(u, a, { awareOfUnicodeTokens: !0 });
  var t = J(i);
  return i && !t && console.warn('A locale object was not found for the provided string ["'.concat(i, '"].')), !t && G() && J(G()) && (t = J(G())), Qe(u, a, { locale: t || null, awareOfUnicodeTokens: !0 });
}
function F(u, a) {
  var i = a.dateFormat, t = a.locale;
  return u && _(u, Array.isArray(i) ? i[0] : i, t) || "";
}
function Xe(u, a) {
  var i = a.hour, t = i === void 0 ? 0 : i, e = a.minute, n = e === void 0 ? 0 : e, s = a.second;
  return de(ue(Rt(u, s === void 0 ? 0 : s), n), t);
}
function K(u, a, i) {
  var t = J(a || G());
  return Lt(u, { locale: t, weekStartsOn: i });
}
function V(u) {
  return ht(u);
}
function ae(u) {
  return mt(u);
}
function Ze(u) {
  return Ft(u);
}
function et() {
  return te(b());
}
function j(u, a) {
  return u && a ? Ht(u, a) : !u && !a;
}
function L(u, a) {
  return u && a ? Qt(u, a) : !u && !a;
}
function ke(u, a) {
  return u && a ? jt(u, a) : !u && !a;
}
function C(u, a) {
  return u && a ? Bt(u, a) : !u && !a;
}
function z(u, a) {
  return u && a ? Kt(u, a) : !u && !a;
}
function he(u, a, i) {
  var t, e = te(a), n = Le(i);
  try {
    t = ce(u, { start: e, end: n });
  } catch {
    t = !1;
  }
  return t;
}
function G() {
  return (typeof window < "u" ? window : globalThis).__localeId__;
}
function J(u) {
  if (typeof u == "string") {
    var a = typeof window < "u" ? window : globalThis;
    return a.__localeData__ ? a.__localeData__[u] : null;
  }
  return u;
}
function Ke(u, a) {
  return _(R(b(), u), "LLLL", a);
}
function yt(u, a) {
  return _(R(b(), u), "LLL", a);
}
function Se(u) {
  var a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, i = a.minDate, t = a.maxDate, e = a.excludeDates, n = a.excludeDateIntervals, s = a.includeDates, r = a.includeDateIntervals, o = a.filterDate;
  return Ce(u, { minDate: i, maxDate: t }) || e && e.some(function(c) {
    return C(u, c);
  }) || n && n.some(function(c) {
    var d = c.start, m = c.end;
    return ce(u, { start: d, end: m });
  }) || s && !s.some(function(c) {
    return C(u, c);
  }) || r && !r.some(function(c) {
    var d = c.start, m = c.end;
    return ce(u, { start: d, end: m });
  }) || o && !o(b(u)) || !1;
}
function Be(u) {
  var a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, i = a.excludeDates, t = a.excludeDateIntervals;
  return t && t.length > 0 ? t.some(function(e) {
    var n = e.start, s = e.end;
    return ce(u, { start: n, end: s });
  }) : i && i.some(function(e) {
    return C(u, e);
  }) || !1;
}
function vt(u) {
  var a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, i = a.minDate, t = a.maxDate, e = a.excludeDates, n = a.includeDates, s = a.filterDate;
  return Ce(u, { minDate: ht(i), maxDate: At(t) }) || e && e.some(function(r) {
    return L(u, r);
  }) || n && !n.some(function(r) {
    return L(u, r);
  }) || s && !s(b(u)) || !1;
}
function Pe(u, a, i, t) {
  var e = w(u), n = I(u), s = w(a), r = I(a), o = w(t);
  return e === s && e === o ? n <= i && i <= r : e < s ? o === e && n <= i || o === s && r >= i || o < s && o > e : void 0;
}
function Zt(u) {
  var a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, i = a.minDate, t = a.maxDate, e = a.excludeDates, n = a.includeDates, s = a.filterDate;
  return Ce(u, { minDate: i, maxDate: t }) || e && e.some(function(r) {
    return ke(u, r);
  }) || n && !n.some(function(r) {
    return ke(u, r);
  }) || s && !s(b(u)) || !1;
}
function Ee(u, a, i) {
  if (!me(a) || !me(i))
    return !1;
  var t = w(a), e = w(i);
  return t <= u && e >= u;
}
function Dt(u) {
  var a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, i = a.minDate, t = a.maxDate, e = a.excludeDates, n = a.includeDates, s = a.filterDate, r = new Date(u, 0, 1);
  return Ce(r, { minDate: mt(i), maxDate: Wt(t) }) || e && e.some(function(o) {
    return j(r, o);
  }) || n && !n.some(function(o) {
    return j(r, o);
  }) || s && !s(b(r)) || !1;
}
function Ne(u, a, i, t) {
  var e = w(u), n = Z(u), s = w(a), r = Z(a), o = w(t);
  return e === s && e === o ? n <= i && i <= r : e < s ? o === e && n <= i || o === s && r >= i || o < s && o > e : void 0;
}
function Ce(u) {
  var a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, i = a.minDate, t = a.maxDate;
  return i && fe(u, i) < 0 || t && fe(u, t) > 0;
}
function tt(u, a) {
  return a.some(function(i) {
    return W(i) === W(u) && A(i) === A(u);
  });
}
function rt(u) {
  var a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, i = a.excludeTimes, t = a.includeTimes, e = a.filterTime;
  return i && tt(u, i) || t && !tt(u, t) || e && !e(u) || !1;
}
function nt(u, a) {
  var i = a.minTime, t = a.maxTime;
  if (!i || !t)
    throw new Error("Both minTime and maxTime props required");
  var e, n = b(), s = de(ue(n, A(u)), W(u)), r = de(ue(n, A(i)), W(i)), o = de(ue(n, A(t)), W(t));
  try {
    e = !ce(s, { start: r, end: o });
  } catch {
    e = !1;
  }
  return e;
}
function ot(u) {
  var a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, i = a.minDate, t = a.includeDates, e = re(u, 1);
  return i && ge(i, e) > 0 || t && t.every(function(n) {
    return ge(n, e) > 0;
  }) || !1;
}
function at(u) {
  var a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, i = a.maxDate, t = a.includeDates, e = q(u, 1);
  return i && ge(e, i) > 0 || t && t.every(function(n) {
    return ge(e, n) > 0;
  }) || !1;
}
function st(u) {
  var a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, i = a.minDate, t = a.includeDates, e = pe(u, 1);
  return i && ye(i, e) > 0 || t && t.every(function(n) {
    return ye(n, e) > 0;
  }) || !1;
}
function it(u) {
  var a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, i = a.maxDate, t = a.includeDates, e = ie(u, 1);
  return i && ye(e, i) > 0 || t && t.every(function(n) {
    return ye(e, n) > 0;
  }) || !1;
}
function kt(u) {
  var a = u.minDate, i = u.includeDates;
  if (i && a) {
    var t = i.filter(function(e) {
      return fe(e, a) >= 0;
    });
    return qe(t);
  }
  return i ? qe(i) : a;
}
function wt(u) {
  var a = u.maxDate, i = u.includeDates;
  if (i && a) {
    var t = i.filter(function(e) {
      return fe(e, a) <= 0;
    });
    return Ue(t);
  }
  return i ? Ue(i) : a;
}
function pt() {
  for (var u = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : [], a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : "react-datepicker__day--highlighted", i = /* @__PURE__ */ new Map(), t = 0, e = u.length; t < e; t++) {
    var n = u[t];
    if (We(n)) {
      var s = _(n, "MM.dd.yyyy"), r = i.get(s) || [];
      r.includes(a) || (r.push(a), i.set(s, r));
    } else if (Fe(n) === "object") {
      var o = Object.keys(n), c = o[0], d = n[o[0]];
      if (typeof c == "string" && d.constructor === Array)
        for (var m = 0, f = d.length; m < f; m++) {
          var y = _(d[m], "MM.dd.yyyy"), g = i.get(y) || [];
          g.includes(c) || (g.push(c), i.set(y, g));
        }
    }
  }
  return i;
}
function er() {
  var u = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : [], a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : "react-datepicker__day--holidays", i = /* @__PURE__ */ new Map();
  return u.forEach(function(t) {
    var e = t.date, n = t.holidayName;
    if (We(e)) {
      var s = _(e, "MM.dd.yyyy"), r = i.get(s) || {};
      if (!("className" in r) || r.className !== a || (o = r.holidayNames, c = [n], o.length !== c.length || !o.every(function(m, f) {
        return m === c[f];
      }))) {
        var o, c;
        r.className = a;
        var d = r.holidayNames;
        r.holidayNames = d ? [].concat(le(d), [n]) : [n], i.set(s, r);
      }
    }
  }), i;
}
function tr(u, a, i, t, e) {
  for (var n = e.length, s = [], r = 0; r < n; r++) {
    var o = Ie(Nt(u, W(e[r])), A(e[r])), c = Ie(u, (i + 1) * t);
    U(o, a) && H(o, c) && s.push(e[r]);
  }
  return s;
}
function ct(u) {
  return u < 10 ? "0".concat(u) : "".concat(u);
}
function ee(u) {
  var a = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : se, i = Math.ceil(w(u) / a) * a;
  return { startPeriod: i - (a - 1), endPeriod: i };
}
function lt(u) {
  var a = u.getSeconds(), i = u.getMilliseconds();
  return ft(u.getTime() - 1e3 * a - i);
}
function rr(u, a, i, t) {
  for (var e = [], n = 0; n < 2 * a + 1; n++) {
    var s = u + a - n, r = !0;
    i && (r = w(i) <= s), t && r && (r = w(t) >= s), r && e.push(s);
  }
  return e;
}
var nr = be(function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i(t) {
    var e;
    E(this, i), p(l(e = a.call(this, t)), "renderOptions", function() {
      var o = e.props.year, c = e.state.yearsList.map(function(f) {
        return h.createElement("div", { className: o === f ? "react-datepicker__year-option react-datepicker__year-option--selected_year" : "react-datepicker__year-option", key: f, onClick: e.onChange.bind(l(e), f), "aria-selected": o === f ? "true" : void 0 }, o === f ? h.createElement("span", { className: "react-datepicker__year-option--selected" }, "✓") : "", f);
      }), d = e.props.minDate ? w(e.props.minDate) : null, m = e.props.maxDate ? w(e.props.maxDate) : null;
      return m && e.state.yearsList.find(function(f) {
        return f === m;
      }) || c.unshift(h.createElement("div", { className: "react-datepicker__year-option", key: "upcoming", onClick: e.incrementYears }, h.createElement("a", { className: "react-datepicker__navigation react-datepicker__navigation--years react-datepicker__navigation--years-upcoming" }))), d && e.state.yearsList.find(function(f) {
        return f === d;
      }) || c.push(h.createElement("div", { className: "react-datepicker__year-option", key: "previous", onClick: e.decrementYears }, h.createElement("a", { className: "react-datepicker__navigation react-datepicker__navigation--years react-datepicker__navigation--years-previous" }))), c;
    }), p(l(e), "onChange", function(o) {
      e.props.onChange(o);
    }), p(l(e), "handleClickOutside", function() {
      e.props.onCancel();
    }), p(l(e), "shiftYears", function(o) {
      var c = e.state.yearsList.map(function(d) {
        return d + o;
      });
      e.setState({ yearsList: c });
    }), p(l(e), "incrementYears", function() {
      return e.shiftYears(1);
    }), p(l(e), "decrementYears", function() {
      return e.shiftYears(-1);
    });
    var n = t.yearDropdownItemNumber, s = t.scrollableYearDropdown, r = n || (s ? 10 : 5);
    return e.state = { yearsList: rr(e.props.year, r, e.props.minDate, e.props.maxDate) }, e.dropdownRef = Et(), e;
  }
  return N(i, [{ key: "componentDidMount", value: function() {
    var t = this.dropdownRef.current;
    if (t) {
      var e = t.children ? Array.from(t.children) : null, n = e ? e.find(function(s) {
        return s.ariaSelected;
      }) : null;
      t.scrollTop = n ? n.offsetTop + (n.clientHeight - t.clientHeight) / 2 : (t.scrollHeight - t.clientHeight) / 2;
    }
  } }, { key: "render", value: function() {
    var t = T({ "react-datepicker__year-dropdown": !0, "react-datepicker__year-dropdown--scrollable": this.props.scrollableYearDropdown });
    return h.createElement("div", { className: t, ref: this.dropdownRef }, this.renderOptions());
  } }]), i;
}()), or = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i() {
    var t;
    E(this, i);
    for (var e = arguments.length, n = new Array(e), s = 0; s < e; s++)
      n[s] = arguments[s];
    return p(l(t = a.call.apply(a, [this].concat(n))), "state", { dropdownVisible: !1 }), p(l(t), "renderSelectOptions", function() {
      for (var r = t.props.minDate ? w(t.props.minDate) : 1900, o = t.props.maxDate ? w(t.props.maxDate) : 2100, c = [], d = r; d <= o; d++)
        c.push(h.createElement("option", { key: d, value: d }, d));
      return c;
    }), p(l(t), "onSelectChange", function(r) {
      t.onChange(r.target.value);
    }), p(l(t), "renderSelectMode", function() {
      return h.createElement("select", { value: t.props.year, className: "react-datepicker__year-select", onChange: t.onSelectChange }, t.renderSelectOptions());
    }), p(l(t), "renderReadView", function(r) {
      return h.createElement("div", { key: "read", style: { visibility: r ? "visible" : "hidden" }, className: "react-datepicker__year-read-view", onClick: function(o) {
        return t.toggleDropdown(o);
      } }, h.createElement("span", { className: "react-datepicker__year-read-view--down-arrow" }), h.createElement("span", { className: "react-datepicker__year-read-view--selected-year" }, t.props.year));
    }), p(l(t), "renderDropdown", function() {
      return h.createElement(nr, { key: "dropdown", year: t.props.year, onChange: t.onChange, onCancel: t.toggleDropdown, minDate: t.props.minDate, maxDate: t.props.maxDate, scrollableYearDropdown: t.props.scrollableYearDropdown, yearDropdownItemNumber: t.props.yearDropdownItemNumber });
    }), p(l(t), "renderScrollMode", function() {
      var r = t.state.dropdownVisible, o = [t.renderReadView(!r)];
      return r && o.unshift(t.renderDropdown()), o;
    }), p(l(t), "onChange", function(r) {
      t.toggleDropdown(), r !== t.props.year && t.props.onChange(r);
    }), p(l(t), "toggleDropdown", function(r) {
      t.setState({ dropdownVisible: !t.state.dropdownVisible }, function() {
        t.props.adjustDateOnChange && t.handleYearChange(t.props.date, r);
      });
    }), p(l(t), "handleYearChange", function(r, o) {
      t.onSelect(r, o), t.setOpen();
    }), p(l(t), "onSelect", function(r, o) {
      t.props.onSelect && t.props.onSelect(r, o);
    }), p(l(t), "setOpen", function() {
      t.props.setOpen && t.props.setOpen(!0);
    }), t;
  }
  return N(i, [{ key: "render", value: function() {
    var t;
    switch (this.props.dropdownMode) {
      case "scroll":
        t = this.renderScrollMode();
        break;
      case "select":
        t = this.renderSelectMode();
    }
    return h.createElement("div", { className: "react-datepicker__year-dropdown-container react-datepicker__year-dropdown-container--".concat(this.props.dropdownMode) }, t);
  } }]), i;
}(), ar = be(function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i() {
    var t;
    E(this, i);
    for (var e = arguments.length, n = new Array(e), s = 0; s < e; s++)
      n[s] = arguments[s];
    return p(l(t = a.call.apply(a, [this].concat(n))), "isSelectedMonth", function(r) {
      return t.props.month === r;
    }), p(l(t), "renderOptions", function() {
      return t.props.monthNames.map(function(r, o) {
        return h.createElement("div", { className: t.isSelectedMonth(o) ? "react-datepicker__month-option react-datepicker__month-option--selected_month" : "react-datepicker__month-option", key: r, onClick: t.onChange.bind(l(t), o), "aria-selected": t.isSelectedMonth(o) ? "true" : void 0 }, t.isSelectedMonth(o) ? h.createElement("span", { className: "react-datepicker__month-option--selected" }, "✓") : "", r);
      });
    }), p(l(t), "onChange", function(r) {
      return t.props.onChange(r);
    }), p(l(t), "handleClickOutside", function() {
      return t.props.onCancel();
    }), t;
  }
  return N(i, [{ key: "render", value: function() {
    return h.createElement("div", { className: "react-datepicker__month-dropdown" }, this.renderOptions());
  } }]), i;
}()), sr = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i() {
    var t;
    E(this, i);
    for (var e = arguments.length, n = new Array(e), s = 0; s < e; s++)
      n[s] = arguments[s];
    return p(l(t = a.call.apply(a, [this].concat(n))), "state", { dropdownVisible: !1 }), p(l(t), "renderSelectOptions", function(r) {
      return r.map(function(o, c) {
        return h.createElement("option", { key: c, value: c }, o);
      });
    }), p(l(t), "renderSelectMode", function(r) {
      return h.createElement("select", { value: t.props.month, className: "react-datepicker__month-select", onChange: function(o) {
        return t.onChange(o.target.value);
      } }, t.renderSelectOptions(r));
    }), p(l(t), "renderReadView", function(r, o) {
      return h.createElement("div", { key: "read", style: { visibility: r ? "visible" : "hidden" }, className: "react-datepicker__month-read-view", onClick: t.toggleDropdown }, h.createElement("span", { className: "react-datepicker__month-read-view--down-arrow" }), h.createElement("span", { className: "react-datepicker__month-read-view--selected-month" }, o[t.props.month]));
    }), p(l(t), "renderDropdown", function(r) {
      return h.createElement(ar, { key: "dropdown", month: t.props.month, monthNames: r, onChange: t.onChange, onCancel: t.toggleDropdown });
    }), p(l(t), "renderScrollMode", function(r) {
      var o = t.state.dropdownVisible, c = [t.renderReadView(!o, r)];
      return o && c.unshift(t.renderDropdown(r)), c;
    }), p(l(t), "onChange", function(r) {
      t.toggleDropdown(), r !== t.props.month && t.props.onChange(r);
    }), p(l(t), "toggleDropdown", function() {
      return t.setState({ dropdownVisible: !t.state.dropdownVisible });
    }), t;
  }
  return N(i, [{ key: "render", value: function() {
    var t, e = this, n = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11].map(this.props.useShortMonthInDropdown ? function(s) {
      return yt(s, e.props.locale);
    } : function(s) {
      return Ke(s, e.props.locale);
    });
    switch (this.props.dropdownMode) {
      case "scroll":
        t = this.renderScrollMode(n);
        break;
      case "select":
        t = this.renderSelectMode(n);
    }
    return h.createElement("div", { className: "react-datepicker__month-dropdown-container react-datepicker__month-dropdown-container--".concat(this.props.dropdownMode) }, t);
  } }]), i;
}();
function ir(u, a) {
  for (var i = [], t = V(u), e = V(a); !U(t, e); )
    i.push(b(t)), t = q(t, 1);
  return i;
}
var pr = be(function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i(t) {
    var e;
    return E(this, i), p(l(e = a.call(this, t)), "renderOptions", function() {
      return e.state.monthYearsList.map(function(n) {
        var s = Re(n), r = j(e.props.date, n) && L(e.props.date, n);
        return h.createElement("div", { className: r ? "react-datepicker__month-year-option--selected_month-year" : "react-datepicker__month-year-option", key: s, onClick: e.onChange.bind(l(e), s), "aria-selected": r ? "true" : void 0 }, r ? h.createElement("span", { className: "react-datepicker__month-year-option--selected" }, "✓") : "", _(n, e.props.dateFormat, e.props.locale));
      });
    }), p(l(e), "onChange", function(n) {
      return e.props.onChange(n);
    }), p(l(e), "handleClickOutside", function() {
      e.props.onCancel();
    }), e.state = { monthYearsList: ir(e.props.minDate, e.props.maxDate) }, e;
  }
  return N(i, [{ key: "render", value: function() {
    var t = T({ "react-datepicker__month-year-dropdown": !0, "react-datepicker__month-year-dropdown--scrollable": this.props.scrollableMonthYearDropdown });
    return h.createElement("div", { className: t }, this.renderOptions());
  } }]), i;
}()), cr = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i() {
    var t;
    E(this, i);
    for (var e = arguments.length, n = new Array(e), s = 0; s < e; s++)
      n[s] = arguments[s];
    return p(l(t = a.call.apply(a, [this].concat(n))), "state", { dropdownVisible: !1 }), p(l(t), "renderSelectOptions", function() {
      for (var r = V(t.props.minDate), o = V(t.props.maxDate), c = []; !U(r, o); ) {
        var d = Re(r);
        c.push(h.createElement("option", { key: d, value: d }, _(r, t.props.dateFormat, t.props.locale))), r = q(r, 1);
      }
      return c;
    }), p(l(t), "onSelectChange", function(r) {
      t.onChange(r.target.value);
    }), p(l(t), "renderSelectMode", function() {
      return h.createElement("select", { value: Re(V(t.props.date)), className: "react-datepicker__month-year-select", onChange: t.onSelectChange }, t.renderSelectOptions());
    }), p(l(t), "renderReadView", function(r) {
      var o = _(t.props.date, t.props.dateFormat, t.props.locale);
      return h.createElement("div", { key: "read", style: { visibility: r ? "visible" : "hidden" }, className: "react-datepicker__month-year-read-view", onClick: function(c) {
        return t.toggleDropdown(c);
      } }, h.createElement("span", { className: "react-datepicker__month-year-read-view--down-arrow" }), h.createElement("span", { className: "react-datepicker__month-year-read-view--selected-month-year" }, o));
    }), p(l(t), "renderDropdown", function() {
      return h.createElement(pr, { key: "dropdown", date: t.props.date, dateFormat: t.props.dateFormat, onChange: t.onChange, onCancel: t.toggleDropdown, minDate: t.props.minDate, maxDate: t.props.maxDate, scrollableMonthYearDropdown: t.props.scrollableMonthYearDropdown, locale: t.props.locale });
    }), p(l(t), "renderScrollMode", function() {
      var r = t.state.dropdownVisible, o = [t.renderReadView(!r)];
      return r && o.unshift(t.renderDropdown()), o;
    }), p(l(t), "onChange", function(r) {
      t.toggleDropdown();
      var o = b(parseInt(r));
      j(t.props.date, o) && L(t.props.date, o) || t.props.onChange(o);
    }), p(l(t), "toggleDropdown", function() {
      return t.setState({ dropdownVisible: !t.state.dropdownVisible });
    }), t;
  }
  return N(i, [{ key: "render", value: function() {
    var t;
    switch (this.props.dropdownMode) {
      case "scroll":
        t = this.renderScrollMode();
        break;
      case "select":
        t = this.renderSelectMode();
    }
    return h.createElement("div", { className: "react-datepicker__month-year-dropdown-container react-datepicker__month-year-dropdown-container--".concat(this.props.dropdownMode) }, t);
  } }]), i;
}(), lr = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i() {
    var t;
    E(this, i);
    for (var e = arguments.length, n = new Array(e), s = 0; s < e; s++)
      n[s] = arguments[s];
    return p(l(t = a.call.apply(a, [this].concat(n))), "dayEl", h.createRef()), p(l(t), "handleClick", function(r) {
      !t.isDisabled() && t.props.onClick && t.props.onClick(r);
    }), p(l(t), "handleMouseEnter", function(r) {
      !t.isDisabled() && t.props.onMouseEnter && t.props.onMouseEnter(r);
    }), p(l(t), "handleOnKeyDown", function(r) {
      r.key === " " && (r.preventDefault(), r.key = "Enter"), t.props.handleOnKeyDown(r);
    }), p(l(t), "isSameDay", function(r) {
      return C(t.props.day, r);
    }), p(l(t), "isKeyboardSelected", function() {
      return !t.props.disabledKeyboardNavigation && !(t.isSameDay(t.props.selected) || t.isSameWeek(t.props.selected)) && (t.isSameDay(t.props.preSelection) || t.isSameWeek(t.props.preSelection));
    }), p(l(t), "isDisabled", function() {
      return Se(t.props.day, t.props);
    }), p(l(t), "isExcluded", function() {
      return Be(t.props.day, t.props);
    }), p(l(t), "isStartOfWeek", function() {
      return C(t.props.day, K(t.props.day, t.props.locale, t.props.calendarStartDay));
    }), p(l(t), "isSameWeek", function(r) {
      return t.props.showWeekPicker && C(r, K(t.props.day, t.props.locale, t.props.calendarStartDay));
    }), p(l(t), "getHighLightedClass", function() {
      var r = t.props, o = r.day, c = r.highlightDates;
      if (!c)
        return !1;
      var d = _(o, "MM.dd.yyyy");
      return c.get(d);
    }), p(l(t), "getHolidaysClass", function() {
      var r = t.props, o = r.day, c = r.holidays;
      if (!c)
        return !1;
      var d = _(o, "MM.dd.yyyy");
      return c.has(d) ? [c.get(d).className] : void 0;
    }), p(l(t), "isInRange", function() {
      var r = t.props, o = r.day, c = r.startDate, d = r.endDate;
      return !(!c || !d) && he(o, c, d);
    }), p(l(t), "isInSelectingRange", function() {
      var r, o = t.props, c = o.day, d = o.selectsStart, m = o.selectsEnd, f = o.selectsRange, y = o.selectsDisabledDaysInRange, g = o.startDate, v = o.endDate, D = (r = t.props.selectingDate) !== null && r !== void 0 ? r : t.props.preSelection;
      return !(!(d || m || f) || !D || !y && t.isDisabled()) && (d && v && (H(D, v) || z(D, v)) ? he(c, D, v) : (m && g && (U(D, g) || z(D, g)) || !(!f || !g || v || !U(D, g) && !z(D, g))) && he(c, g, D));
    }), p(l(t), "isSelectingRangeStart", function() {
      var r;
      if (!t.isInSelectingRange())
        return !1;
      var o = t.props, c = o.day, d = o.startDate, m = o.selectsStart, f = (r = t.props.selectingDate) !== null && r !== void 0 ? r : t.props.preSelection;
      return C(c, m ? f : d);
    }), p(l(t), "isSelectingRangeEnd", function() {
      var r;
      if (!t.isInSelectingRange())
        return !1;
      var o = t.props, c = o.day, d = o.endDate, m = o.selectsEnd, f = o.selectsRange, y = (r = t.props.selectingDate) !== null && r !== void 0 ? r : t.props.preSelection;
      return C(c, m || f ? y : d);
    }), p(l(t), "isRangeStart", function() {
      var r = t.props, o = r.day, c = r.startDate, d = r.endDate;
      return !(!c || !d) && C(c, o);
    }), p(l(t), "isRangeEnd", function() {
      var r = t.props, o = r.day, c = r.startDate, d = r.endDate;
      return !(!c || !d) && C(d, o);
    }), p(l(t), "isWeekend", function() {
      var r = It(t.props.day);
      return r === 0 || r === 6;
    }), p(l(t), "isAfterMonth", function() {
      return t.props.month !== void 0 && (t.props.month + 1) % 12 === I(t.props.day);
    }), p(l(t), "isBeforeMonth", function() {
      return t.props.month !== void 0 && (I(t.props.day) + 1) % 12 === t.props.month;
    }), p(l(t), "isCurrentDay", function() {
      return t.isSameDay(b());
    }), p(l(t), "isSelected", function() {
      return t.isSameDay(t.props.selected) || t.isSameWeek(t.props.selected);
    }), p(l(t), "getClassNames", function(r) {
      var o, c = t.props.dayClassName ? t.props.dayClassName(r) : void 0;
      return T("react-datepicker__day", c, "react-datepicker__day--" + _(t.props.day, "ddd", o), { "react-datepicker__day--disabled": t.isDisabled(), "react-datepicker__day--excluded": t.isExcluded(), "react-datepicker__day--selected": t.isSelected(), "react-datepicker__day--keyboard-selected": t.isKeyboardSelected(), "react-datepicker__day--range-start": t.isRangeStart(), "react-datepicker__day--range-end": t.isRangeEnd(), "react-datepicker__day--in-range": t.isInRange(), "react-datepicker__day--in-selecting-range": t.isInSelectingRange(), "react-datepicker__day--selecting-range-start": t.isSelectingRangeStart(), "react-datepicker__day--selecting-range-end": t.isSelectingRangeEnd(), "react-datepicker__day--today": t.isCurrentDay(), "react-datepicker__day--weekend": t.isWeekend(), "react-datepicker__day--outside-month": t.isAfterMonth() || t.isBeforeMonth() }, t.getHighLightedClass("react-datepicker__day--highlighted"), t.getHolidaysClass());
    }), p(l(t), "getAriaLabel", function() {
      var r = t.props, o = r.day, c = r.ariaLabelPrefixWhenEnabled, d = c === void 0 ? "Choose" : c, m = r.ariaLabelPrefixWhenDisabled, f = m === void 0 ? "Not available" : m, y = t.isDisabled() || t.isExcluded() ? f : d;
      return "".concat(y, " ").concat(_(o, "PPPP", t.props.locale));
    }), p(l(t), "getTitle", function() {
      var r = t.props, o = r.day, c = r.holidays, d = c === void 0 ? /* @__PURE__ */ new Map() : c, m = _(o, "MM.dd.yyyy");
      return d.has(m) && d.get(m).holidayNames.length > 0 ? d.get(m).holidayNames.join(", ") : "";
    }), p(l(t), "getTabIndex", function(r, o) {
      var c = r || t.props.selected, d = o || t.props.preSelection;
      return (!t.props.showWeekPicker || !t.props.showWeekNumber && t.isStartOfWeek()) && (t.isKeyboardSelected() || t.isSameDay(c) && C(d, c)) ? 0 : -1;
    }), p(l(t), "handleFocusDay", function() {
      var r, o = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {}, c = !1;
      t.getTabIndex() === 0 && !o.isInputFocused && t.isSameDay(t.props.preSelection) && (document.activeElement && document.activeElement !== document.body || (c = !0), t.props.inline && !t.props.shouldFocusDayInline && (c = !1), t.props.containerRef && t.props.containerRef.current && t.props.containerRef.current.contains(document.activeElement) && document.activeElement.classList.contains("react-datepicker__day") && (c = !0), t.props.monthShowsDuplicateDaysEnd && t.isAfterMonth() && (c = !1), t.props.monthShowsDuplicateDaysStart && t.isBeforeMonth() && (c = !1)), c && ((r = t.dayEl.current) === null || r === void 0 || r.focus({ preventScroll: !0 }));
    }), p(l(t), "renderDayContents", function() {
      return t.props.monthShowsDuplicateDaysEnd && t.isAfterMonth() || t.props.monthShowsDuplicateDaysStart && t.isBeforeMonth() ? null : t.props.renderDayContents ? t.props.renderDayContents(Ve(t.props.day), t.props.day) : Ve(t.props.day);
    }), p(l(t), "render", function() {
      return h.createElement("div", { ref: t.dayEl, className: t.getClassNames(t.props.day), onKeyDown: t.handleOnKeyDown, onClick: t.handleClick, onMouseEnter: t.handleMouseEnter, tabIndex: t.getTabIndex(), "aria-label": t.getAriaLabel(), role: "option", title: t.getTitle(), "aria-disabled": t.isDisabled(), "aria-current": t.isCurrentDay() ? "date" : void 0, "aria-selected": t.isSelected() || t.isInRange() }, t.renderDayContents(), t.getTitle() !== "" && h.createElement("span", { className: "holiday-overlay" }, t.getTitle()));
    }), t;
  }
  return N(i, [{ key: "componentDidMount", value: function() {
    this.handleFocusDay();
  } }, { key: "componentDidUpdate", value: function(t) {
    this.handleFocusDay(t);
  } }]), i;
}(), ur = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i() {
    var t;
    E(this, i);
    for (var e = arguments.length, n = new Array(e), s = 0; s < e; s++)
      n[s] = arguments[s];
    return p(l(t = a.call.apply(a, [this].concat(n))), "weekNumberEl", h.createRef()), p(l(t), "handleClick", function(r) {
      t.props.onClick && t.props.onClick(r);
    }), p(l(t), "handleOnKeyDown", function(r) {
      r.key === " " && (r.preventDefault(), r.key = "Enter"), t.props.handleOnKeyDown(r);
    }), p(l(t), "isKeyboardSelected", function() {
      return !t.props.disabledKeyboardNavigation && !C(t.props.date, t.props.selected) && C(t.props.date, t.props.preSelection);
    }), p(l(t), "getTabIndex", function() {
      return t.props.showWeekPicker && t.props.showWeekNumber && (t.isKeyboardSelected() || C(t.props.date, t.props.selected) && C(t.props.preSelection, t.props.selected)) ? 0 : -1;
    }), p(l(t), "handleFocusWeekNumber", function() {
      var r = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {}, o = !1;
      t.getTabIndex() === 0 && !r.isInputFocused && C(t.props.date, t.props.preSelection) && (document.activeElement && document.activeElement !== document.body || (o = !0), t.props.inline && !t.props.shouldFocusDayInline && (o = !1), t.props.containerRef && t.props.containerRef.current && t.props.containerRef.current.contains(document.activeElement) && document.activeElement && document.activeElement.classList.contains("react-datepicker__week-number") && (o = !0)), o && t.weekNumberEl.current && t.weekNumberEl.current.focus({ preventScroll: !0 });
    }), t;
  }
  return N(i, [{ key: "componentDidMount", value: function() {
    this.handleFocusWeekNumber();
  } }, { key: "componentDidUpdate", value: function(t) {
    this.handleFocusWeekNumber(t);
  } }, { key: "render", value: function() {
    var t = this.props, e = t.weekNumber, n = t.ariaLabelPrefix, s = n === void 0 ? "week " : n, r = { "react-datepicker__week-number": !0, "react-datepicker__week-number--clickable": !!t.onClick, "react-datepicker__week-number--selected": C(this.props.date, this.props.selected), "react-datepicker__week-number--keyboard-selected": this.isKeyboardSelected() };
    return h.createElement("div", { ref: this.weekNumberEl, className: T(r), "aria-label": "".concat(s, " ").concat(this.props.weekNumber), onClick: this.handleClick, onKeyDown: this.handleOnKeyDown, tabIndex: this.getTabIndex() }, e);
  } }], [{ key: "defaultProps", get: function() {
    return { ariaLabelPrefix: "week " };
  } }]), i;
}(), dr = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i() {
    var t;
    E(this, i);
    for (var e = arguments.length, n = new Array(e), s = 0; s < e; s++)
      n[s] = arguments[s];
    return p(l(t = a.call.apply(a, [this].concat(n))), "handleDayClick", function(r, o) {
      t.props.onDayClick && t.props.onDayClick(r, o);
    }), p(l(t), "handleDayMouseEnter", function(r) {
      t.props.onDayMouseEnter && t.props.onDayMouseEnter(r);
    }), p(l(t), "handleWeekClick", function(r, o, c) {
      if (typeof t.props.onWeekSelect == "function" && t.props.onWeekSelect(r, o, c), t.props.showWeekPicker) {
        var d = K(r, t.props.locale, t.props.calendarStartDay);
        t.handleDayClick(d, c);
      }
      t.props.shouldCloseOnSelect && t.props.setOpen(!1);
    }), p(l(t), "formatWeekNumber", function(r) {
      return t.props.formatWeekNumber ? t.props.formatWeekNumber(r) : function(o, c) {
        var d = G() && J(G());
        return Ot(o, d ? { locale: d } : null);
      }(r);
    }), p(l(t), "renderDays", function() {
      var r = K(t.props.day, t.props.locale, t.props.calendarStartDay), o = [], c = t.formatWeekNumber(r);
      if (t.props.showWeekNumber) {
        var d = t.props.onWeekSelect || t.props.showWeekPicker ? t.handleWeekClick.bind(l(t), r, c) : void 0;
        o.push(h.createElement(ur, { key: "W", weekNumber: c, date: r, onClick: d, selected: t.props.selected, preSelection: t.props.preSelection, ariaLabelPrefix: t.props.ariaLabelPrefix, showWeekPicker: t.props.showWeekPicker, showWeekNumber: t.props.showWeekNumber, disabledKeyboardNavigation: t.props.disabledKeyboardNavigation, handleOnKeyDown: t.props.handleOnKeyDown, isInputFocused: t.props.isInputFocused, containerRef: t.props.containerRef }));
      }
      return o.concat([0, 1, 2, 3, 4, 5, 6].map(function(m) {
        var f = we(r, m);
        return h.createElement(lr, { ariaLabelPrefixWhenEnabled: t.props.chooseDayAriaLabelPrefix, ariaLabelPrefixWhenDisabled: t.props.disabledDayAriaLabelPrefix, key: f.valueOf(), day: f, month: t.props.month, onClick: t.handleDayClick.bind(l(t), f), onMouseEnter: t.handleDayMouseEnter.bind(l(t), f), minDate: t.props.minDate, maxDate: t.props.maxDate, excludeDates: t.props.excludeDates, excludeDateIntervals: t.props.excludeDateIntervals, includeDates: t.props.includeDates, includeDateIntervals: t.props.includeDateIntervals, highlightDates: t.props.highlightDates, holidays: t.props.holidays, selectingDate: t.props.selectingDate, filterDate: t.props.filterDate, preSelection: t.props.preSelection, selected: t.props.selected, selectsStart: t.props.selectsStart, selectsEnd: t.props.selectsEnd, selectsRange: t.props.selectsRange, showWeekPicker: t.props.showWeekPicker, showWeekNumber: t.props.showWeekNumber, selectsDisabledDaysInRange: t.props.selectsDisabledDaysInRange, startDate: t.props.startDate, endDate: t.props.endDate, dayClassName: t.props.dayClassName, renderDayContents: t.props.renderDayContents, disabledKeyboardNavigation: t.props.disabledKeyboardNavigation, handleOnKeyDown: t.props.handleOnKeyDown, isInputFocused: t.props.isInputFocused, containerRef: t.props.containerRef, inline: t.props.inline, shouldFocusDayInline: t.props.shouldFocusDayInline, monthShowsDuplicateDaysEnd: t.props.monthShowsDuplicateDaysEnd, monthShowsDuplicateDaysStart: t.props.monthShowsDuplicateDaysStart, locale: t.props.locale });
      }));
    }), p(l(t), "startOfWeek", function() {
      return K(t.props.day, t.props.locale, t.props.calendarStartDay);
    }), p(l(t), "isKeyboardSelected", function() {
      return !t.props.disabledKeyboardNavigation && !C(t.startOfWeek(), t.props.selected) && C(t.startOfWeek(), t.props.preSelection);
    }), t;
  }
  return N(i, [{ key: "render", value: function() {
    var t = { "react-datepicker__week": !0, "react-datepicker__week--selected": C(this.startOfWeek(), this.props.selected), "react-datepicker__week--keyboard-selected": this.isKeyboardSelected() };
    return h.createElement("div", { className: T(t) }, this.renderDays());
  } }], [{ key: "defaultProps", get: function() {
    return { shouldCloseOnSelect: !0 };
  } }]), i;
}(), bt = "two_columns", St = "three_columns", Ct = "four_columns", xe = p(p(p({}, bt, { grid: [[0, 1], [2, 3], [4, 5], [6, 7], [8, 9], [10, 11]], verticalNavigationOffset: 2 }), St, { grid: [[0, 1, 2], [3, 4, 5], [6, 7, 8], [9, 10, 11]], verticalNavigationOffset: 3 }), Ct, { grid: [[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11]], verticalNavigationOffset: 4 });
function ut(u, a) {
  return u ? Ct : a ? bt : St;
}
var hr = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i() {
    var t;
    E(this, i);
    for (var e = arguments.length, n = new Array(e), s = 0; s < e; s++)
      n[s] = arguments[s];
    return p(l(t = a.call.apply(a, [this].concat(n))), "MONTH_REFS", le(Array(12)).map(function() {
      return h.createRef();
    })), p(l(t), "QUARTER_REFS", le(Array(4)).map(function() {
      return h.createRef();
    })), p(l(t), "isDisabled", function(r) {
      return Se(r, t.props);
    }), p(l(t), "isExcluded", function(r) {
      return Be(r, t.props);
    }), p(l(t), "handleDayClick", function(r, o) {
      t.props.onDayClick && t.props.onDayClick(r, o, t.props.orderInDisplay);
    }), p(l(t), "handleDayMouseEnter", function(r) {
      t.props.onDayMouseEnter && t.props.onDayMouseEnter(r);
    }), p(l(t), "handleMouseLeave", function() {
      t.props.onMouseLeave && t.props.onMouseLeave();
    }), p(l(t), "isRangeStartMonth", function(r) {
      var o = t.props, c = o.day, d = o.startDate, m = o.endDate;
      return !(!d || !m) && L(R(c, r), d);
    }), p(l(t), "isRangeStartQuarter", function(r) {
      var o = t.props, c = o.day, d = o.startDate, m = o.endDate;
      return !(!d || !m) && ke(X(c, r), d);
    }), p(l(t), "isRangeEndMonth", function(r) {
      var o = t.props, c = o.day, d = o.startDate, m = o.endDate;
      return !(!d || !m) && L(R(c, r), m);
    }), p(l(t), "isRangeEndQuarter", function(r) {
      var o = t.props, c = o.day, d = o.startDate, m = o.endDate;
      return !(!d || !m) && ke(X(c, r), m);
    }), p(l(t), "isInSelectingRangeMonth", function(r) {
      var o, c = t.props, d = c.day, m = c.selectsStart, f = c.selectsEnd, y = c.selectsRange, g = c.startDate, v = c.endDate, D = (o = t.props.selectingDate) !== null && o !== void 0 ? o : t.props.preSelection;
      return !(!(m || f || y) || !D) && (m && v ? Pe(D, v, r, d) : (f && g || !(!y || !g || v)) && Pe(g, D, r, d));
    }), p(l(t), "isSelectingMonthRangeStart", function(r) {
      var o;
      if (!t.isInSelectingRangeMonth(r))
        return !1;
      var c = t.props, d = c.day, m = c.startDate, f = c.selectsStart, y = R(d, r), g = (o = t.props.selectingDate) !== null && o !== void 0 ? o : t.props.preSelection;
      return L(y, f ? g : m);
    }), p(l(t), "isSelectingMonthRangeEnd", function(r) {
      var o;
      if (!t.isInSelectingRangeMonth(r))
        return !1;
      var c = t.props, d = c.day, m = c.endDate, f = c.selectsEnd, y = c.selectsRange, g = R(d, r), v = (o = t.props.selectingDate) !== null && o !== void 0 ? o : t.props.preSelection;
      return L(g, f || y ? v : m);
    }), p(l(t), "isInSelectingRangeQuarter", function(r) {
      var o, c = t.props, d = c.day, m = c.selectsStart, f = c.selectsEnd, y = c.selectsRange, g = c.startDate, v = c.endDate, D = (o = t.props.selectingDate) !== null && o !== void 0 ? o : t.props.preSelection;
      return !(!(m || f || y) || !D) && (m && v ? Ne(D, v, r, d) : (f && g || !(!y || !g || v)) && Ne(g, D, r, d));
    }), p(l(t), "isWeekInMonth", function(r) {
      var o = t.props.day, c = we(r, 6);
      return L(r, o) || L(c, o);
    }), p(l(t), "isCurrentMonth", function(r, o) {
      return w(r) === w(b()) && o === I(b());
    }), p(l(t), "isCurrentQuarter", function(r, o) {
      return w(r) === w(b()) && o === Z(b());
    }), p(l(t), "isSelectedMonth", function(r, o, c) {
      return I(c) === o && w(r) === w(c);
    }), p(l(t), "isSelectedQuarter", function(r, o, c) {
      return Z(r) === o && w(r) === w(c);
    }), p(l(t), "renderWeeks", function() {
      for (var r = [], o = t.props.fixedHeight, c = 0, d = !1, m = K(V(t.props.day), t.props.locale, t.props.calendarStartDay); r.push(h.createElement(dr, { ariaLabelPrefix: t.props.weekAriaLabelPrefix, chooseDayAriaLabelPrefix: t.props.chooseDayAriaLabelPrefix, disabledDayAriaLabelPrefix: t.props.disabledDayAriaLabelPrefix, key: c, day: m, month: I(t.props.day), onDayClick: t.handleDayClick, onDayMouseEnter: t.handleDayMouseEnter, onWeekSelect: t.props.onWeekSelect, formatWeekNumber: t.props.formatWeekNumber, locale: t.props.locale, minDate: t.props.minDate, maxDate: t.props.maxDate, excludeDates: t.props.excludeDates, excludeDateIntervals: t.props.excludeDateIntervals, includeDates: t.props.includeDates, includeDateIntervals: t.props.includeDateIntervals, inline: t.props.inline, shouldFocusDayInline: t.props.shouldFocusDayInline, highlightDates: t.props.highlightDates, holidays: t.props.holidays, selectingDate: t.props.selectingDate, filterDate: t.props.filterDate, preSelection: t.props.preSelection, selected: t.props.selected, selectsStart: t.props.selectsStart, selectsEnd: t.props.selectsEnd, selectsRange: t.props.selectsRange, selectsDisabledDaysInRange: t.props.selectsDisabledDaysInRange, showWeekNumber: t.props.showWeekNumbers, showWeekPicker: t.props.showWeekPicker, startDate: t.props.startDate, endDate: t.props.endDate, dayClassName: t.props.dayClassName, setOpen: t.props.setOpen, shouldCloseOnSelect: t.props.shouldCloseOnSelect, disabledKeyboardNavigation: t.props.disabledKeyboardNavigation, renderDayContents: t.props.renderDayContents, handleOnKeyDown: t.props.handleOnKeyDown, isInputFocused: t.props.isInputFocused, containerRef: t.props.containerRef, calendarStartDay: t.props.calendarStartDay, monthShowsDuplicateDaysEnd: t.props.monthShowsDuplicateDaysEnd, monthShowsDuplicateDaysStart: t.props.monthShowsDuplicateDaysStart })), !d; ) {
        c++, m = Oe(m, 1);
        var f = o && c >= 6, y = !o && !t.isWeekInMonth(m);
        if (f || y) {
          if (!t.props.peekNextMonth)
            break;
          d = !0;
        }
      }
      return r;
    }), p(l(t), "onMonthClick", function(r, o) {
      t.handleDayClick(V(R(t.props.day, o)), r);
    }), p(l(t), "onMonthMouseEnter", function(r) {
      t.handleDayMouseEnter(V(R(t.props.day, r)));
    }), p(l(t), "handleMonthNavigation", function(r, o) {
      t.isDisabled(o) || t.isExcluded(o) || (t.props.setPreSelection(o), t.MONTH_REFS[r].current && t.MONTH_REFS[r].current.focus());
    }), p(l(t), "onMonthKeyDown", function(r, o) {
      var c = t.props, d = c.selected, m = c.preSelection, f = c.disabledKeyboardNavigation, y = c.showTwoColumnMonthYearPicker, g = c.showFourColumnMonthYearPicker, v = c.setPreSelection, D = r.key;
      if (D !== "Tab" && r.preventDefault(), !f) {
        var M = ut(g, y), S = xe[M].verticalNavigationOffset, k = xe[M].grid;
        switch (D) {
          case "Enter":
            t.onMonthClick(r, o), v(d);
            break;
          case "ArrowRight":
            t.handleMonthNavigation(o === 11 ? 0 : o + 1, q(m, 1));
            break;
          case "ArrowLeft":
            t.handleMonthNavigation(o === 0 ? 11 : o - 1, re(m, 1));
            break;
          case "ArrowUp":
            t.handleMonthNavigation(k[0].includes(o) ? o + 12 - S : o - S, re(m, S));
            break;
          case "ArrowDown":
            t.handleMonthNavigation(k[k.length - 1].includes(o) ? o - 12 + S : o + S, q(m, S));
        }
      }
    }), p(l(t), "onQuarterClick", function(r, o) {
      t.handleDayClick(Ze(X(t.props.day, o)), r);
    }), p(l(t), "onQuarterMouseEnter", function(r) {
      t.handleDayMouseEnter(Ze(X(t.props.day, r)));
    }), p(l(t), "handleQuarterNavigation", function(r, o) {
      t.isDisabled(o) || t.isExcluded(o) || (t.props.setPreSelection(o), t.QUARTER_REFS[r - 1].current && t.QUARTER_REFS[r - 1].current.focus());
    }), p(l(t), "onQuarterKeyDown", function(r, o) {
      var c = r.key;
      if (!t.props.disabledKeyboardNavigation)
        switch (c) {
          case "Enter":
            t.onQuarterClick(r, o), t.props.setPreSelection(t.props.selected);
            break;
          case "ArrowRight":
            t.handleQuarterNavigation(o === 4 ? 1 : o + 1, xt(t.props.preSelection, 1));
            break;
          case "ArrowLeft":
            t.handleQuarterNavigation(o === 1 ? 4 : o - 1, Tt(t.props.preSelection, 1));
        }
    }), p(l(t), "getMonthClassNames", function(r) {
      var o = t.props, c = o.day, d = o.startDate, m = o.endDate, f = o.selected, y = o.minDate, g = o.maxDate, v = o.preSelection, D = o.monthClassName, M = o.excludeDates, S = o.includeDates, k = D ? D(R(c, r)) : void 0, P = R(c, r);
      return T("react-datepicker__month-text", "react-datepicker__month-".concat(r), k, { "react-datepicker__month-text--disabled": (y || g || M || S) && vt(P, t.props), "react-datepicker__month-text--selected": t.isSelectedMonth(c, r, f), "react-datepicker__month-text--keyboard-selected": !t.props.disabledKeyboardNavigation && I(v) === r, "react-datepicker__month-text--in-selecting-range": t.isInSelectingRangeMonth(r), "react-datepicker__month-text--in-range": Pe(d, m, r, c), "react-datepicker__month-text--range-start": t.isRangeStartMonth(r), "react-datepicker__month-text--range-end": t.isRangeEndMonth(r), "react-datepicker__month-text--selecting-range-start": t.isSelectingMonthRangeStart(r), "react-datepicker__month-text--selecting-range-end": t.isSelectingMonthRangeEnd(r), "react-datepicker__month-text--today": t.isCurrentMonth(c, r) });
    }), p(l(t), "getTabIndex", function(r) {
      var o = I(t.props.preSelection);
      return t.props.disabledKeyboardNavigation || r !== o ? "-1" : "0";
    }), p(l(t), "getQuarterTabIndex", function(r) {
      var o = Z(t.props.preSelection);
      return t.props.disabledKeyboardNavigation || r !== o ? "-1" : "0";
    }), p(l(t), "getAriaLabel", function(r) {
      var o = t.props, c = o.chooseDayAriaLabelPrefix, d = c === void 0 ? "Choose" : c, m = o.disabledDayAriaLabelPrefix, f = m === void 0 ? "Not available" : m, y = o.day, g = R(y, r), v = t.isDisabled(g) || t.isExcluded(g) ? f : d;
      return "".concat(v, " ").concat(_(g, "MMMM yyyy"));
    }), p(l(t), "getQuarterClassNames", function(r) {
      var o = t.props, c = o.day, d = o.startDate, m = o.endDate, f = o.selected, y = o.minDate, g = o.maxDate, v = o.preSelection, D = o.disabledKeyboardNavigation;
      return T("react-datepicker__quarter-text", "react-datepicker__quarter-".concat(r), { "react-datepicker__quarter-text--disabled": (y || g) && Zt(X(c, r), t.props), "react-datepicker__quarter-text--selected": t.isSelectedQuarter(c, r, f), "react-datepicker__quarter-text--keyboard-selected": !D && Z(v) === r, "react-datepicker__quarter-text--in-selecting-range": t.isInSelectingRangeQuarter(r), "react-datepicker__quarter-text--in-range": Ne(d, m, r, c), "react-datepicker__quarter-text--range-start": t.isRangeStartQuarter(r), "react-datepicker__quarter-text--range-end": t.isRangeEndQuarter(r) });
    }), p(l(t), "getMonthContent", function(r) {
      var o = t.props, c = o.showFullMonthYearPicker, d = o.renderMonthContent, m = o.locale, f = o.day, y = yt(r, m), g = Ke(r, m);
      return d ? d(r, y, g, f) : c ? g : y;
    }), p(l(t), "getQuarterContent", function(r) {
      var o = t.props, c = o.renderQuarterContent, d = function(m, f) {
        return _(X(b(), m), "QQQ", f);
      }(r, o.locale);
      return c ? c(r, d) : d;
    }), p(l(t), "renderMonths", function() {
      var r = t.props, o = r.showTwoColumnMonthYearPicker, c = r.showFourColumnMonthYearPicker, d = r.day, m = r.selected;
      return xe[ut(c, o)].grid.map(function(f, y) {
        return h.createElement("div", { className: "react-datepicker__month-wrapper", key: y }, f.map(function(g, v) {
          return h.createElement("div", { ref: t.MONTH_REFS[g], key: v, onClick: function(D) {
            t.onMonthClick(D, g);
          }, onKeyDown: function(D) {
            t.onMonthKeyDown(D, g);
          }, onMouseEnter: function() {
            return t.onMonthMouseEnter(g);
          }, tabIndex: t.getTabIndex(g), className: t.getMonthClassNames(g), role: "option", "aria-label": t.getAriaLabel(g), "aria-current": t.isCurrentMonth(d, g) ? "date" : void 0, "aria-selected": t.isSelectedMonth(d, g, m) }, t.getMonthContent(g));
        }));
      });
    }), p(l(t), "renderQuarters", function() {
      var r = t.props, o = r.day, c = r.selected;
      return h.createElement("div", { className: "react-datepicker__quarter-wrapper" }, [1, 2, 3, 4].map(function(d, m) {
        return h.createElement("div", { key: m, ref: t.QUARTER_REFS[m], role: "option", onClick: function(f) {
          t.onQuarterClick(f, d);
        }, onKeyDown: function(f) {
          t.onQuarterKeyDown(f, d);
        }, onMouseEnter: function() {
          return t.onQuarterMouseEnter(d);
        }, className: t.getQuarterClassNames(d), "aria-selected": t.isSelectedQuarter(o, d, c), tabIndex: t.getQuarterTabIndex(d), "aria-current": t.isCurrentQuarter(o, d) ? "date" : void 0 }, t.getQuarterContent(d));
      }));
    }), p(l(t), "getClassNames", function() {
      var r = t.props, o = r.selectingDate, c = r.selectsStart, d = r.selectsEnd, m = r.showMonthYearPicker, f = r.showQuarterYearPicker, y = r.showWeekPicker;
      return T("react-datepicker__month", { "react-datepicker__month--selecting-range": o && (c || d) }, { "react-datepicker__monthPicker": m }, { "react-datepicker__quarterPicker": f }, { "react-datepicker__weekPicker": y });
    }), t;
  }
  return N(i, [{ key: "render", value: function() {
    var t = this.props, e = t.showMonthYearPicker, n = t.showQuarterYearPicker, s = t.day, r = t.ariaLabelPrefix, o = r === void 0 ? "month " : r;
    return h.createElement("div", { className: this.getClassNames(), onMouseLeave: this.handleMouseLeave, "aria-label": "".concat(o, " ").concat(_(s, "yyyy-MM")), role: "listbox" }, e ? this.renderMonths() : n ? this.renderQuarters() : this.renderWeeks());
  } }]), i;
}(), _t = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i() {
    var t;
    E(this, i);
    for (var e = arguments.length, n = new Array(e), s = 0; s < e; s++)
      n[s] = arguments[s];
    return p(l(t = a.call.apply(a, [this].concat(n))), "state", { height: null }), p(l(t), "scrollToTheSelectedTime", function() {
      requestAnimationFrame(function() {
        t.list && (t.list.scrollTop = t.centerLi && i.calcCenterPosition(t.props.monthRef ? t.props.monthRef.clientHeight - t.header.clientHeight : t.list.clientHeight, t.centerLi));
      });
    }), p(l(t), "handleClick", function(r) {
      (t.props.minTime || t.props.maxTime) && nt(r, t.props) || (t.props.excludeTimes || t.props.includeTimes || t.props.filterTime) && rt(r, t.props) || t.props.onChange(r);
    }), p(l(t), "isSelectedTime", function(r) {
      return t.props.selected && (o = t.props.selected, c = r, lt(o).getTime() === lt(c).getTime());
      var o, c;
    }), p(l(t), "isDisabledTime", function(r) {
      return (t.props.minTime || t.props.maxTime) && nt(r, t.props) || (t.props.excludeTimes || t.props.includeTimes || t.props.filterTime) && rt(r, t.props);
    }), p(l(t), "liClasses", function(r) {
      var o = ["react-datepicker__time-list-item", t.props.timeClassName ? t.props.timeClassName(r) : void 0];
      return t.isSelectedTime(r) && o.push("react-datepicker__time-list-item--selected"), t.isDisabledTime(r) && o.push("react-datepicker__time-list-item--disabled"), t.props.injectTimes && (60 * W(r) + A(r)) % t.props.intervals != 0 && o.push("react-datepicker__time-list-item--injected"), o.join(" ");
    }), p(l(t), "handleOnKeyDown", function(r, o) {
      r.key === " " && (r.preventDefault(), r.key = "Enter"), r.key !== "ArrowUp" && r.key !== "ArrowLeft" || !r.target.previousSibling || (r.preventDefault(), r.target.previousSibling.focus()), r.key !== "ArrowDown" && r.key !== "ArrowRight" || !r.target.nextSibling || (r.preventDefault(), r.target.nextSibling.focus()), r.key === "Enter" && t.handleClick(o), t.props.handleOnKeyDown(r);
    }), p(l(t), "renderTimes", function() {
      for (var r = [], o = t.props.format ? t.props.format : "p", c = t.props.intervals, d = t.props.selected || t.props.openToDate || b(), m = te(d), f = t.props.injectTimes && t.props.injectTimes.sort(function(k, P) {
        return k - P;
      }), y = 60 * function(k) {
        var P = new Date(k.getFullYear(), k.getMonth(), k.getDate()), O = new Date(k.getFullYear(), k.getMonth(), k.getDate(), 24);
        return Math.round((+O - +P) / 36e5);
      }(d), g = y / c, v = 0; v < g; v++) {
        var D = Ie(m, v * c);
        if (r.push(D), f) {
          var M = tr(m, D, v, c, f);
          r = r.concat(M);
        }
      }
      var S = r.reduce(function(k, P) {
        return P.getTime() <= d.getTime() ? P : k;
      }, r[0]);
      return r.map(function(k, P) {
        return h.createElement("li", { key: P, onClick: t.handleClick.bind(l(t), k), className: t.liClasses(k), ref: function(O) {
          k === S && (t.centerLi = O);
        }, onKeyDown: function(O) {
          t.handleOnKeyDown(O, k);
        }, tabIndex: k === S ? 0 : -1, role: "option", "aria-selected": t.isSelectedTime(k) ? "true" : void 0, "aria-disabled": t.isDisabledTime(k) ? "true" : void 0 }, _(k, o, t.props.locale));
      });
    }), t;
  }
  return N(i, [{ key: "componentDidMount", value: function() {
    this.scrollToTheSelectedTime(), this.props.monthRef && this.header && this.setState({ height: this.props.monthRef.clientHeight - this.header.clientHeight });
  } }, { key: "render", value: function() {
    var t = this, e = this.state.height;
    return h.createElement("div", { className: "react-datepicker__time-container ".concat(this.props.todayButton ? "react-datepicker__time-container--with-today-button" : "") }, h.createElement("div", { className: "react-datepicker__header react-datepicker__header--time ".concat(this.props.showTimeSelectOnly ? "react-datepicker__header--time--only" : ""), ref: function(n) {
      t.header = n;
    } }, h.createElement("div", { className: "react-datepicker-time__header" }, this.props.timeCaption)), h.createElement("div", { className: "react-datepicker__time" }, h.createElement("div", { className: "react-datepicker__time-box" }, h.createElement("ul", { className: "react-datepicker__time-list", ref: function(n) {
      t.list = n;
    }, style: e ? { height: e } : {}, role: "listbox", "aria-label": this.props.timeCaption }, this.renderTimes()))));
  } }], [{ key: "defaultProps", get: function() {
    return { intervals: 30, onTimeChange: function() {
    }, todayButton: null, timeCaption: "Time" };
  } }]), i;
}();
p(_t, "calcCenterPosition", function(u, a) {
  return a.offsetTop - (u / 2 - a.clientHeight / 2);
});
var mr = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i(t) {
    var e;
    return E(this, i), p(l(e = a.call(this, t)), "YEAR_REFS", le(Array(e.props.yearItemNumber)).map(function() {
      return h.createRef();
    })), p(l(e), "isDisabled", function(n) {
      return Se(n, e.props);
    }), p(l(e), "isExcluded", function(n) {
      return Be(n, e.props);
    }), p(l(e), "selectingDate", function() {
      var n;
      return (n = e.props.selectingDate) !== null && n !== void 0 ? n : e.props.preSelection;
    }), p(l(e), "updateFocusOnPaginate", function(n) {
      var s = (function() {
        this.YEAR_REFS[n].current.focus();
      }).bind(l(e));
      window.requestAnimationFrame(s);
    }), p(l(e), "handleYearClick", function(n, s) {
      e.props.onDayClick && e.props.onDayClick(n, s);
    }), p(l(e), "handleYearNavigation", function(n, s) {
      var r = e.props, o = r.date, c = r.yearItemNumber, d = ee(o, c).startPeriod;
      e.isDisabled(s) || e.isExcluded(s) || (e.props.setPreSelection(s), n - d == -1 ? e.updateFocusOnPaginate(c - 1) : n - d === c ? e.updateFocusOnPaginate(0) : e.YEAR_REFS[n - d].current.focus());
    }), p(l(e), "isSameDay", function(n, s) {
      return C(n, s);
    }), p(l(e), "isCurrentYear", function(n) {
      return n === w(b());
    }), p(l(e), "isRangeStart", function(n) {
      return e.props.startDate && e.props.endDate && j(B(b(), n), e.props.startDate);
    }), p(l(e), "isRangeEnd", function(n) {
      return e.props.startDate && e.props.endDate && j(B(b(), n), e.props.endDate);
    }), p(l(e), "isInRange", function(n) {
      return Ee(n, e.props.startDate, e.props.endDate);
    }), p(l(e), "isInSelectingRange", function(n) {
      var s = e.props, r = s.selectsStart, o = s.selectsEnd, c = s.selectsRange, d = s.startDate, m = s.endDate;
      return !(!(r || o || c) || !e.selectingDate()) && (r && m ? Ee(n, e.selectingDate(), m) : (o && d || !(!c || !d || m)) && Ee(n, d, e.selectingDate()));
    }), p(l(e), "isSelectingRangeStart", function(n) {
      if (!e.isInSelectingRange(n))
        return !1;
      var s = e.props, r = s.startDate, o = s.selectsStart, c = B(b(), n);
      return j(c, o ? e.selectingDate() : r);
    }), p(l(e), "isSelectingRangeEnd", function(n) {
      if (!e.isInSelectingRange(n))
        return !1;
      var s = e.props, r = s.endDate, o = s.selectsEnd, c = s.selectsRange, d = B(b(), n);
      return j(d, o || c ? e.selectingDate() : r);
    }), p(l(e), "isKeyboardSelected", function(n) {
      var s = ae(B(e.props.date, n));
      return !e.props.disabledKeyboardNavigation && !e.props.inline && !C(s, ae(e.props.selected)) && C(s, ae(e.props.preSelection));
    }), p(l(e), "onYearClick", function(n, s) {
      var r = e.props.date;
      e.handleYearClick(ae(B(r, s)), n);
    }), p(l(e), "onYearKeyDown", function(n, s) {
      var r = n.key;
      if (!e.props.disabledKeyboardNavigation)
        switch (r) {
          case "Enter":
            e.onYearClick(n, s), e.props.setPreSelection(e.props.selected);
            break;
          case "ArrowRight":
            e.handleYearNavigation(s + 1, ie(e.props.preSelection, 1));
            break;
          case "ArrowLeft":
            e.handleYearNavigation(s - 1, pe(e.props.preSelection, 1));
        }
    }), p(l(e), "getYearClassNames", function(n) {
      var s = e.props, r = s.minDate, o = s.maxDate, c = s.selected, d = s.excludeDates, m = s.includeDates, f = s.filterDate;
      return T("react-datepicker__year-text", { "react-datepicker__year-text--selected": n === w(c), "react-datepicker__year-text--disabled": (r || o || d || m || f) && Dt(n, e.props), "react-datepicker__year-text--keyboard-selected": e.isKeyboardSelected(n), "react-datepicker__year-text--range-start": e.isRangeStart(n), "react-datepicker__year-text--range-end": e.isRangeEnd(n), "react-datepicker__year-text--in-range": e.isInRange(n), "react-datepicker__year-text--in-selecting-range": e.isInSelectingRange(n), "react-datepicker__year-text--selecting-range-start": e.isSelectingRangeStart(n), "react-datepicker__year-text--selecting-range-end": e.isSelectingRangeEnd(n), "react-datepicker__year-text--today": e.isCurrentYear(n) });
    }), p(l(e), "getYearTabIndex", function(n) {
      return e.props.disabledKeyboardNavigation ? "-1" : n === w(e.props.preSelection) ? "0" : "-1";
    }), p(l(e), "getYearContainerClassNames", function() {
      var n = e.props, s = n.selectingDate, r = n.selectsStart, o = n.selectsEnd, c = n.selectsRange;
      return T("react-datepicker__year", { "react-datepicker__year--selecting-range": s && (r || o || c) });
    }), p(l(e), "getYearContent", function(n) {
      return e.props.renderYearContent ? e.props.renderYearContent(n) : n;
    }), e;
  }
  return N(i, [{ key: "render", value: function() {
    for (var t = this, e = [], n = this.props, s = n.date, r = n.yearItemNumber, o = n.onYearMouseEnter, c = n.onYearMouseLeave, d = ee(s, r), m = d.startPeriod, f = d.endPeriod, y = function(v) {
      e.push(h.createElement("div", { ref: t.YEAR_REFS[v - m], onClick: function(D) {
        t.onYearClick(D, v);
      }, onKeyDown: function(D) {
        t.onYearKeyDown(D, v);
      }, tabIndex: t.getYearTabIndex(v), className: t.getYearClassNames(v), onMouseEnter: function(D) {
        return o(D, v);
      }, onMouseLeave: function(D) {
        return c(D, v);
      }, key: v, "aria-current": t.isCurrentYear(v) ? "date" : void 0 }, t.getYearContent(v)));
    }, g = m; g <= f; g++)
      y(g);
    return h.createElement("div", { className: this.getYearContainerClassNames() }, h.createElement("div", { className: "react-datepicker__year-wrapper", onMouseLeave: this.props.clearSelectingDate }, e));
  } }]), i;
}(), fr = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i(t) {
    var e;
    return E(this, i), p(l(e = a.call(this, t)), "onTimeChange", function(n) {
      e.setState({ time: n });
      var s = e.props.date, r = s instanceof Date && !isNaN(s) ? s : /* @__PURE__ */ new Date();
      r.setHours(n.split(":")[0]), r.setMinutes(n.split(":")[1]), e.props.onChange(r);
    }), p(l(e), "renderTimeInput", function() {
      var n = e.state.time, s = e.props, r = s.date, o = s.timeString, c = s.customTimeInput;
      return c ? h.cloneElement(c, { date: r, value: n, onChange: e.onTimeChange }) : h.createElement("input", { type: "time", className: "react-datepicker-time__input", placeholder: "Time", name: "time-input", required: !0, value: n, onChange: function(d) {
        e.onTimeChange(d.target.value || o);
      } });
    }), e.state = { time: e.props.timeString }, e;
  }
  return N(i, [{ key: "render", value: function() {
    return h.createElement("div", { className: "react-datepicker__input-time-container" }, h.createElement("div", { className: "react-datepicker-time__caption" }, this.props.timeInputLabel), h.createElement("div", { className: "react-datepicker-time__input-container" }, h.createElement("div", { className: "react-datepicker-time__input" }, this.renderTimeInput())));
  } }], [{ key: "getDerivedStateFromProps", value: function(t, e) {
    return t.timeString !== e.time ? { time: t.timeString } : null;
  } }]), i;
}();
function gr(u) {
  var a = u.className, i = u.children, t = u.showPopperArrow, e = u.arrowProps, n = e === void 0 ? {} : e;
  return h.createElement("div", { className: a }, t && h.createElement("div", ne({ className: "react-datepicker__triangle" }, n)), i);
}
var yr = ["react-datepicker__year-select", "react-datepicker__month-select", "react-datepicker__month-year-select"], vr = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i(t) {
    var e;
    return E(this, i), p(l(e = a.call(this, t)), "handleClickOutside", function(n) {
      e.props.onClickOutside(n);
    }), p(l(e), "setClickOutsideRef", function() {
      return e.containerRef.current;
    }), p(l(e), "handleDropdownFocus", function(n) {
      (function() {
        var s = ((arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {}).className || "").split(/\s+/);
        return yr.some(function(r) {
          return s.indexOf(r) >= 0;
        });
      })(n.target) && e.props.onDropdownFocus();
    }), p(l(e), "getDateInView", function() {
      var n = e.props, s = n.preSelection, r = n.selected, o = n.openToDate, c = kt(e.props), d = wt(e.props), m = b(), f = o || r || s;
      return f || (c && H(m, c) ? c : d && U(m, d) ? d : m);
    }), p(l(e), "increaseMonth", function() {
      e.setState(function(n) {
        var s = n.date;
        return { date: q(s, 1) };
      }, function() {
        return e.handleMonthChange(e.state.date);
      });
    }), p(l(e), "decreaseMonth", function() {
      e.setState(function(n) {
        var s = n.date;
        return { date: re(s, 1) };
      }, function() {
        return e.handleMonthChange(e.state.date);
      });
    }), p(l(e), "handleDayClick", function(n, s, r) {
      e.props.onSelect(n, s, r), e.props.setPreSelection && e.props.setPreSelection(n);
    }), p(l(e), "handleDayMouseEnter", function(n) {
      e.setState({ selectingDate: n }), e.props.onDayMouseEnter && e.props.onDayMouseEnter(n);
    }), p(l(e), "handleMonthMouseLeave", function() {
      e.setState({ selectingDate: null }), e.props.onMonthMouseLeave && e.props.onMonthMouseLeave();
    }), p(l(e), "handleYearMouseEnter", function(n, s) {
      e.setState({ selectingDate: B(b(), s) }), e.props.onYearMouseEnter && e.props.onYearMouseEnter(n, s);
    }), p(l(e), "handleYearMouseLeave", function(n, s) {
      e.props.onYearMouseLeave && e.props.onYearMouseLeave(n, s);
    }), p(l(e), "handleYearChange", function(n) {
      e.props.onYearChange && (e.props.onYearChange(n), e.setState({ isRenderAriaLiveMessage: !0 })), e.props.adjustDateOnChange && (e.props.onSelect && e.props.onSelect(n), e.props.setOpen && e.props.setOpen(!0)), e.props.setPreSelection && e.props.setPreSelection(n);
    }), p(l(e), "handleMonthChange", function(n) {
      e.handleCustomMonthChange(n), e.props.adjustDateOnChange && (e.props.onSelect && e.props.onSelect(n), e.props.setOpen && e.props.setOpen(!0)), e.props.setPreSelection && e.props.setPreSelection(n);
    }), p(l(e), "handleCustomMonthChange", function(n) {
      e.props.onMonthChange && (e.props.onMonthChange(n), e.setState({ isRenderAriaLiveMessage: !0 }));
    }), p(l(e), "handleMonthYearChange", function(n) {
      e.handleYearChange(n), e.handleMonthChange(n);
    }), p(l(e), "changeYear", function(n) {
      e.setState(function(s) {
        var r = s.date;
        return { date: B(r, n) };
      }, function() {
        return e.handleYearChange(e.state.date);
      });
    }), p(l(e), "changeMonth", function(n) {
      e.setState(function(s) {
        var r = s.date;
        return { date: R(r, n) };
      }, function() {
        return e.handleMonthChange(e.state.date);
      });
    }), p(l(e), "changeMonthYear", function(n) {
      e.setState(function(s) {
        var r = s.date;
        return { date: B(R(r, I(n)), w(n)) };
      }, function() {
        return e.handleMonthYearChange(e.state.date);
      });
    }), p(l(e), "header", function() {
      var n = K(arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : e.state.date, e.props.locale, e.props.calendarStartDay), s = [];
      return e.props.showWeekNumbers && s.push(h.createElement("div", { key: "W", className: "react-datepicker__day-name" }, e.props.weekLabel || "#")), s.concat([0, 1, 2, 3, 4, 5, 6].map(function(r) {
        var o = we(n, r), c = e.formatWeekday(o, e.props.locale), d = e.props.weekDayClassName ? e.props.weekDayClassName(o) : void 0;
        return h.createElement("div", { key: r, className: T("react-datepicker__day-name", d) }, c);
      }));
    }), p(l(e), "formatWeekday", function(n, s) {
      return e.props.formatWeekDay ? function(r, o, c) {
        return o(_(r, "EEEE", c));
      }(n, e.props.formatWeekDay, s) : e.props.useWeekdaysShort ? function(r, o) {
        return _(r, "EEE", o);
      }(n, s) : function(r, o) {
        return _(r, "EEEEEE", o);
      }(n, s);
    }), p(l(e), "decreaseYear", function() {
      e.setState(function(n) {
        var s = n.date;
        return { date: pe(s, e.props.showYearPicker ? e.props.yearItemNumber : 1) };
      }, function() {
        return e.handleYearChange(e.state.date);
      });
    }), p(l(e), "clearSelectingDate", function() {
      e.setState({ selectingDate: null });
    }), p(l(e), "renderPreviousButton", function() {
      if (!e.props.renderCustomHeader) {
        var n;
        switch (!0) {
          case e.props.showMonthYearPicker:
            n = st(e.state.date, e.props);
            break;
          case e.props.showYearPicker:
            n = function(M) {
              var S = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, k = S.minDate, P = S.yearItemNumber, O = P === void 0 ? se : P, oe = ee(ae(pe(M, O)), O).endPeriod, $ = k && w(k);
              return $ && $ > oe || !1;
            }(e.state.date, e.props);
            break;
          default:
            n = ot(e.state.date, e.props);
        }
        if ((e.props.forceShowMonthNavigation || e.props.showDisabledMonthNavigation || !n) && !e.props.showTimeSelectOnly) {
          var s = ["react-datepicker__navigation", "react-datepicker__navigation--previous"], r = e.decreaseMonth;
          (e.props.showMonthYearPicker || e.props.showQuarterYearPicker || e.props.showYearPicker) && (r = e.decreaseYear), n && e.props.showDisabledMonthNavigation && (s.push("react-datepicker__navigation--previous--disabled"), r = null);
          var o = e.props.showMonthYearPicker || e.props.showQuarterYearPicker || e.props.showYearPicker, c = e.props, d = c.previousMonthButtonLabel, m = c.previousYearButtonLabel, f = e.props, y = f.previousMonthAriaLabel, g = y === void 0 ? typeof d == "string" ? d : "Previous Month" : y, v = f.previousYearAriaLabel, D = v === void 0 ? typeof m == "string" ? m : "Previous Year" : v;
          return h.createElement("button", { type: "button", className: s.join(" "), onClick: r, onKeyDown: e.props.handleOnKeyDown, "aria-label": o ? D : g }, h.createElement("span", { className: ["react-datepicker__navigation-icon", "react-datepicker__navigation-icon--previous"].join(" ") }, o ? e.props.previousYearButtonLabel : e.props.previousMonthButtonLabel));
        }
      }
    }), p(l(e), "increaseYear", function() {
      e.setState(function(n) {
        var s = n.date;
        return { date: ie(s, e.props.showYearPicker ? e.props.yearItemNumber : 1) };
      }, function() {
        return e.handleYearChange(e.state.date);
      });
    }), p(l(e), "renderNextButton", function() {
      if (!e.props.renderCustomHeader) {
        var n;
        switch (!0) {
          case e.props.showMonthYearPicker:
            n = it(e.state.date, e.props);
            break;
          case e.props.showYearPicker:
            n = function(M) {
              var S = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, k = S.maxDate, P = S.yearItemNumber, O = P === void 0 ? se : P, oe = ee(ie(M, O), O).startPeriod, $ = k && w(k);
              return $ && $ < oe || !1;
            }(e.state.date, e.props);
            break;
          default:
            n = at(e.state.date, e.props);
        }
        if ((e.props.forceShowMonthNavigation || e.props.showDisabledMonthNavigation || !n) && !e.props.showTimeSelectOnly) {
          var s = ["react-datepicker__navigation", "react-datepicker__navigation--next"];
          e.props.showTimeSelect && s.push("react-datepicker__navigation--next--with-time"), e.props.todayButton && s.push("react-datepicker__navigation--next--with-today-button");
          var r = e.increaseMonth;
          (e.props.showMonthYearPicker || e.props.showQuarterYearPicker || e.props.showYearPicker) && (r = e.increaseYear), n && e.props.showDisabledMonthNavigation && (s.push("react-datepicker__navigation--next--disabled"), r = null);
          var o = e.props.showMonthYearPicker || e.props.showQuarterYearPicker || e.props.showYearPicker, c = e.props, d = c.nextMonthButtonLabel, m = c.nextYearButtonLabel, f = e.props, y = f.nextMonthAriaLabel, g = y === void 0 ? typeof d == "string" ? d : "Next Month" : y, v = f.nextYearAriaLabel, D = v === void 0 ? typeof m == "string" ? m : "Next Year" : v;
          return h.createElement("button", { type: "button", className: s.join(" "), onClick: r, onKeyDown: e.props.handleOnKeyDown, "aria-label": o ? D : g }, h.createElement("span", { className: ["react-datepicker__navigation-icon", "react-datepicker__navigation-icon--next"].join(" ") }, o ? e.props.nextYearButtonLabel : e.props.nextMonthButtonLabel));
        }
      }
    }), p(l(e), "renderCurrentMonth", function() {
      var n = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : e.state.date, s = ["react-datepicker__current-month"];
      return e.props.showYearDropdown && s.push("react-datepicker__current-month--hasYearDropdown"), e.props.showMonthDropdown && s.push("react-datepicker__current-month--hasMonthDropdown"), e.props.showMonthYearDropdown && s.push("react-datepicker__current-month--hasMonthYearDropdown"), h.createElement("div", { className: s.join(" ") }, _(n, e.props.dateFormat, e.props.locale));
    }), p(l(e), "renderYearDropdown", function() {
      var n = arguments.length > 0 && arguments[0] !== void 0 && arguments[0];
      if (e.props.showYearDropdown && !n)
        return h.createElement(or, { adjustDateOnChange: e.props.adjustDateOnChange, date: e.state.date, onSelect: e.props.onSelect, setOpen: e.props.setOpen, dropdownMode: e.props.dropdownMode, onChange: e.changeYear, minDate: e.props.minDate, maxDate: e.props.maxDate, year: w(e.state.date), scrollableYearDropdown: e.props.scrollableYearDropdown, yearDropdownItemNumber: e.props.yearDropdownItemNumber });
    }), p(l(e), "renderMonthDropdown", function() {
      var n = arguments.length > 0 && arguments[0] !== void 0 && arguments[0];
      if (e.props.showMonthDropdown && !n)
        return h.createElement(sr, { dropdownMode: e.props.dropdownMode, locale: e.props.locale, onChange: e.changeMonth, month: I(e.state.date), useShortMonthInDropdown: e.props.useShortMonthInDropdown });
    }), p(l(e), "renderMonthYearDropdown", function() {
      var n = arguments.length > 0 && arguments[0] !== void 0 && arguments[0];
      if (e.props.showMonthYearDropdown && !n)
        return h.createElement(cr, { dropdownMode: e.props.dropdownMode, locale: e.props.locale, dateFormat: e.props.dateFormat, onChange: e.changeMonthYear, minDate: e.props.minDate, maxDate: e.props.maxDate, date: e.state.date, scrollableMonthYearDropdown: e.props.scrollableMonthYearDropdown });
    }), p(l(e), "handleTodayButtonClick", function(n) {
      e.props.onSelect(et(), n), e.props.setPreSelection && e.props.setPreSelection(et());
    }), p(l(e), "renderTodayButton", function() {
      if (e.props.todayButton && !e.props.showTimeSelectOnly)
        return h.createElement("div", { className: "react-datepicker__today-button", onClick: function(n) {
          return e.handleTodayButtonClick(n);
        } }, e.props.todayButton);
    }), p(l(e), "renderDefaultHeader", function(n) {
      var s = n.monthDate, r = n.i;
      return h.createElement("div", { className: "react-datepicker__header ".concat(e.props.showTimeSelect ? "react-datepicker__header--has-time-select" : "") }, e.renderCurrentMonth(s), h.createElement("div", { className: "react-datepicker__header__dropdown react-datepicker__header__dropdown--".concat(e.props.dropdownMode), onFocus: e.handleDropdownFocus }, e.renderMonthDropdown(r !== 0), e.renderMonthYearDropdown(r !== 0), e.renderYearDropdown(r !== 0)), h.createElement("div", { className: "react-datepicker__day-names" }, e.header(s)));
    }), p(l(e), "renderCustomHeader", function() {
      var n = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {}, s = n.monthDate, r = n.i;
      if (e.props.showTimeSelect && !e.state.monthContainer || e.props.showTimeSelectOnly)
        return null;
      var o = ot(e.state.date, e.props), c = at(e.state.date, e.props), d = st(e.state.date, e.props), m = it(e.state.date, e.props), f = !e.props.showMonthYearPicker && !e.props.showQuarterYearPicker && !e.props.showYearPicker;
      return h.createElement("div", { className: "react-datepicker__header react-datepicker__header--custom", onFocus: e.props.onDropdownFocus }, e.props.renderCustomHeader(ve(ve({}, e.state), {}, { customHeaderCount: r, monthDate: s, changeMonth: e.changeMonth, changeYear: e.changeYear, decreaseMonth: e.decreaseMonth, increaseMonth: e.increaseMonth, decreaseYear: e.decreaseYear, increaseYear: e.increaseYear, prevMonthButtonDisabled: o, nextMonthButtonDisabled: c, prevYearButtonDisabled: d, nextYearButtonDisabled: m })), f && h.createElement("div", { className: "react-datepicker__day-names" }, e.header(s)));
    }), p(l(e), "renderYearHeader", function() {
      var n = e.state.date, s = e.props, r = s.showYearPicker, o = ee(n, s.yearItemNumber), c = o.startPeriod, d = o.endPeriod;
      return h.createElement("div", { className: "react-datepicker__header react-datepicker-year-header" }, r ? "".concat(c, " - ").concat(d) : w(n));
    }), p(l(e), "renderHeader", function(n) {
      switch (!0) {
        case e.props.renderCustomHeader !== void 0:
          return e.renderCustomHeader(n);
        case (e.props.showMonthYearPicker || e.props.showQuarterYearPicker || e.props.showYearPicker):
          return e.renderYearHeader(n);
        default:
          return e.renderDefaultHeader(n);
      }
    }), p(l(e), "renderMonths", function() {
      var n;
      if (!e.props.showTimeSelectOnly && !e.props.showYearPicker) {
        for (var s = [], r = e.props.showPreviousMonths ? e.props.monthsShown - 1 : 0, o = re(e.state.date, r), c = (n = e.props.monthSelectedIn) !== null && n !== void 0 ? n : r, d = 0; d < e.props.monthsShown; ++d) {
          var m = q(o, d - c + r), f = "month-".concat(d), y = d < e.props.monthsShown - 1, g = d > 0;
          s.push(h.createElement("div", { key: f, ref: function(v) {
            e.monthContainer = v;
          }, className: "react-datepicker__month-container" }, e.renderHeader({ monthDate: m, i: d }), h.createElement(hr, { chooseDayAriaLabelPrefix: e.props.chooseDayAriaLabelPrefix, disabledDayAriaLabelPrefix: e.props.disabledDayAriaLabelPrefix, weekAriaLabelPrefix: e.props.weekAriaLabelPrefix, ariaLabelPrefix: e.props.monthAriaLabelPrefix, onChange: e.changeMonthYear, day: m, dayClassName: e.props.dayClassName, calendarStartDay: e.props.calendarStartDay, monthClassName: e.props.monthClassName, onDayClick: e.handleDayClick, handleOnKeyDown: e.props.handleOnDayKeyDown, onDayMouseEnter: e.handleDayMouseEnter, onMouseLeave: e.handleMonthMouseLeave, onWeekSelect: e.props.onWeekSelect, orderInDisplay: d, formatWeekNumber: e.props.formatWeekNumber, locale: e.props.locale, minDate: e.props.minDate, maxDate: e.props.maxDate, excludeDates: e.props.excludeDates, excludeDateIntervals: e.props.excludeDateIntervals, highlightDates: e.props.highlightDates, holidays: e.props.holidays, selectingDate: e.state.selectingDate, includeDates: e.props.includeDates, includeDateIntervals: e.props.includeDateIntervals, inline: e.props.inline, shouldFocusDayInline: e.props.shouldFocusDayInline, fixedHeight: e.props.fixedHeight, filterDate: e.props.filterDate, preSelection: e.props.preSelection, setPreSelection: e.props.setPreSelection, selected: e.props.selected, selectsStart: e.props.selectsStart, selectsEnd: e.props.selectsEnd, selectsRange: e.props.selectsRange, selectsDisabledDaysInRange: e.props.selectsDisabledDaysInRange, showWeekNumbers: e.props.showWeekNumbers, startDate: e.props.startDate, endDate: e.props.endDate, peekNextMonth: e.props.peekNextMonth, setOpen: e.props.setOpen, shouldCloseOnSelect: e.props.shouldCloseOnSelect, renderDayContents: e.props.renderDayContents, renderMonthContent: e.props.renderMonthContent, renderQuarterContent: e.props.renderQuarterContent, renderYearContent: e.props.renderYearContent, disabledKeyboardNavigation: e.props.disabledKeyboardNavigation, showMonthYearPicker: e.props.showMonthYearPicker, showFullMonthYearPicker: e.props.showFullMonthYearPicker, showTwoColumnMonthYearPicker: e.props.showTwoColumnMonthYearPicker, showFourColumnMonthYearPicker: e.props.showFourColumnMonthYearPicker, showYearPicker: e.props.showYearPicker, showQuarterYearPicker: e.props.showQuarterYearPicker, showWeekPicker: e.props.showWeekPicker, isInputFocused: e.props.isInputFocused, containerRef: e.containerRef, monthShowsDuplicateDaysEnd: y, monthShowsDuplicateDaysStart: g })));
        }
        return s;
      }
    }), p(l(e), "renderYears", function() {
      if (!e.props.showTimeSelectOnly)
        return e.props.showYearPicker ? h.createElement("div", { className: "react-datepicker__year--container" }, e.renderHeader(), h.createElement(mr, ne({ onDayClick: e.handleDayClick, selectingDate: e.state.selectingDate, clearSelectingDate: e.clearSelectingDate, date: e.state.date }, e.props, { onYearMouseEnter: e.handleYearMouseEnter, onYearMouseLeave: e.handleYearMouseLeave }))) : void 0;
    }), p(l(e), "renderTimeSection", function() {
      if (e.props.showTimeSelect && (e.state.monthContainer || e.props.showTimeSelectOnly))
        return h.createElement(_t, { selected: e.props.selected, openToDate: e.props.openToDate, onChange: e.props.onTimeChange, timeClassName: e.props.timeClassName, format: e.props.timeFormat, includeTimes: e.props.includeTimes, intervals: e.props.timeIntervals, minTime: e.props.minTime, maxTime: e.props.maxTime, excludeTimes: e.props.excludeTimes, filterTime: e.props.filterTime, timeCaption: e.props.timeCaption, todayButton: e.props.todayButton, showMonthDropdown: e.props.showMonthDropdown, showMonthYearDropdown: e.props.showMonthYearDropdown, showYearDropdown: e.props.showYearDropdown, withPortal: e.props.withPortal, monthRef: e.state.monthContainer, injectTimes: e.props.injectTimes, locale: e.props.locale, handleOnKeyDown: e.props.handleOnKeyDown, showTimeSelectOnly: e.props.showTimeSelectOnly });
    }), p(l(e), "renderInputTimeSection", function() {
      var n = new Date(e.props.selected), s = Q(n) && e.props.selected ? "".concat(ct(n.getHours()), ":").concat(ct(n.getMinutes())) : "";
      if (e.props.showTimeInput)
        return h.createElement(fr, { date: n, timeString: s, timeInputLabel: e.props.timeInputLabel, onChange: e.props.onTimeChange, customTimeInput: e.props.customTimeInput });
    }), p(l(e), "renderAriaLiveRegion", function() {
      var n, s = ee(e.state.date, e.props.yearItemNumber), r = s.startPeriod, o = s.endPeriod;
      return n = e.props.showYearPicker ? "".concat(r, " - ").concat(o) : e.props.showMonthYearPicker || e.props.showQuarterYearPicker ? w(e.state.date) : "".concat(Ke(I(e.state.date), e.props.locale), " ").concat(w(e.state.date)), h.createElement("span", { role: "alert", "aria-live": "polite", className: "react-datepicker__aria-live" }, e.state.isRenderAriaLiveMessage && n);
    }), p(l(e), "renderChildren", function() {
      if (e.props.children)
        return h.createElement("div", { className: "react-datepicker__children-container" }, e.props.children);
    }), e.containerRef = h.createRef(), e.state = { date: e.getDateInView(), selectingDate: null, monthContainer: null, isRenderAriaLiveMessage: !1 }, e;
  }
  return N(i, [{ key: "componentDidMount", value: function() {
    var t = this;
    this.props.showTimeSelect && (this.assignMonthContainer = void t.setState({ monthContainer: t.monthContainer }));
  } }, { key: "componentDidUpdate", value: function(t) {
    var e = this;
    if (!this.props.preSelection || C(this.props.preSelection, t.preSelection) && this.props.monthSelectedIn === t.monthSelectedIn)
      this.props.openToDate && !C(this.props.openToDate, t.openToDate) && this.setState({ date: this.props.openToDate });
    else {
      var n = !L(this.state.date, this.props.preSelection);
      this.setState({ date: this.props.preSelection }, function() {
        return n && e.handleCustomMonthChange(e.state.date);
      });
    }
  } }, { key: "render", value: function() {
    var t = this.props.container || gr;
    return h.createElement("div", { style: { display: "contents" }, ref: this.containerRef }, h.createElement(t, { className: T("react-datepicker", this.props.className, { "react-datepicker--time-only": this.props.showTimeSelectOnly }), showPopperArrow: this.props.showPopperArrow, arrowProps: this.props.arrowProps }, this.renderAriaLiveRegion(), this.renderPreviousButton(), this.renderNextButton(), this.renderMonths(), this.renderYears(), this.renderTodayButton(), this.renderTimeSection(), this.renderInputTimeSection(), this.renderChildren()));
  } }], [{ key: "defaultProps", get: function() {
    return { onDropdownFocus: function() {
    }, monthsShown: 1, forceShowMonthNavigation: !1, timeCaption: "Time", previousYearButtonLabel: "Previous Year", nextYearButtonLabel: "Next Year", previousMonthButtonLabel: "Previous Month", nextMonthButtonLabel: "Next Month", customTimeInput: null, yearItemNumber: se };
  } }]), i;
}(), Dr = function(u) {
  var a = u.icon, i = u.className, t = i === void 0 ? "" : i, e = u.onClick, n = "react-datepicker__calendar-icon";
  return h.isValidElement(a) ? h.cloneElement(a, { className: "".concat(a.props.className || "", " ").concat(n, " ").concat(t), onClick: function(s) {
    typeof a.props.onClick == "function" && a.props.onClick(s), typeof e == "function" && e(s);
  } }) : typeof a == "string" ? h.createElement("i", { className: "".concat(n, " ").concat(a, " ").concat(t), "aria-hidden": "true", onClick: e }) : h.createElement("svg", { className: "".concat(n, " ").concat(t), xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 448 512", onClick: e }, h.createElement("path", { d: "M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z" }));
}, Mt = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i(t) {
    var e;
    return E(this, i), (e = a.call(this, t)).el = document.createElement("div"), e;
  }
  return N(i, [{ key: "componentDidMount", value: function() {
    this.portalRoot = (this.props.portalHost || document).getElementById(this.props.portalId), this.portalRoot || (this.portalRoot = document.createElement("div"), this.portalRoot.setAttribute("id", this.props.portalId), (this.props.portalHost || document.body).appendChild(this.portalRoot)), this.portalRoot.appendChild(this.el);
  } }, { key: "componentWillUnmount", value: function() {
    this.portalRoot.removeChild(this.el);
  } }, { key: "render", value: function() {
    return qt.createPortal(this.props.children, this.el);
  } }]), i;
}(), kr = function(u) {
  return !u.disabled && u.tabIndex !== -1;
}, Pt = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i(t) {
    var e;
    return E(this, i), p(l(e = a.call(this, t)), "getTabChildren", function() {
      return Array.prototype.slice.call(e.tabLoopRef.current.querySelectorAll("[tabindex], a, button, input, select, textarea"), 1, -1).filter(kr);
    }), p(l(e), "handleFocusStart", function() {
      var n = e.getTabChildren();
      n && n.length > 1 && n[n.length - 1].focus();
    }), p(l(e), "handleFocusEnd", function() {
      var n = e.getTabChildren();
      n && n.length > 1 && n[0].focus();
    }), e.tabLoopRef = h.createRef(), e;
  }
  return N(i, [{ key: "render", value: function() {
    return this.props.enableTabLoop ? h.createElement("div", { className: "react-datepicker__tab-loop", ref: this.tabLoopRef }, h.createElement("div", { className: "react-datepicker__tab-loop__start", tabIndex: "0", onFocus: this.handleFocusStart }), this.props.children, h.createElement("div", { className: "react-datepicker__tab-loop__end", tabIndex: "0", onFocus: this.handleFocusEnd })) : this.props.children;
  } }], [{ key: "defaultProps", get: function() {
    return { enableTabLoop: !0 };
  } }]), i;
}(), wr = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i() {
    return E(this, i), a.apply(this, arguments);
  }
  return N(i, [{ key: "render", value: function() {
    var t, e = this.props, n = e.className, s = e.wrapperClassName, r = e.hidePopper, o = e.popperComponent, c = e.popperModifiers, d = e.popperPlacement, m = e.popperProps, f = e.targetComponent, y = e.enableTabLoop, g = e.popperOnKeyDown, v = e.portalId, D = e.portalHost;
    if (!r) {
      var M = T("react-datepicker-popper", n);
      t = h.createElement(Ut, ne({ modifiers: c, placement: d }, m), function(k) {
        var P = k.ref, O = k.style, oe = k.placement, $ = k.arrowProps;
        return h.createElement(Pt, { enableTabLoop: y }, h.createElement("div", { ref: P, style: O, className: M, "data-placement": oe, onKeyDown: g }, h.cloneElement(o, { arrowProps: $ })));
      });
    }
    this.props.popperContainer && (t = h.createElement(this.props.popperContainer, {}, t)), v && !r && (t = h.createElement(Mt, { portalId: v, portalHost: D }, t));
    var S = T("react-datepicker-wrapper", s);
    return h.createElement($t, { className: "react-datepicker-manager" }, h.createElement(zt, null, function(k) {
      var P = k.ref;
      return h.createElement("div", { ref: P, className: S }, f);
    }), t);
  } }], [{ key: "defaultProps", get: function() {
    return { hidePopper: !0, popperModifiers: [], popperProps: {}, popperPlacement: "bottom-start" };
  } }]), i;
}(), dt = "react-datepicker-ignore-onclickoutside", br = be(vr), Ye = "Date input not valid.", On = function(u) {
  x(i, h.Component);
  var a = Y(i);
  function i(t) {
    var e;
    return E(this, i), p(l(e = a.call(this, t)), "getPreSelection", function() {
      return e.props.openToDate ? e.props.openToDate : e.props.selectsEnd && e.props.startDate ? e.props.startDate : e.props.selectsStart && e.props.endDate ? e.props.endDate : b();
    }), p(l(e), "modifyHolidays", function() {
      var n;
      return (n = e.props.holidays) === null || n === void 0 ? void 0 : n.reduce(function(s, r) {
        var o = new Date(r.date);
        return me(o) ? [].concat(le(s), [ve(ve({}, r), {}, { date: o })]) : s;
      }, []);
    }), p(l(e), "calcInitialState", function() {
      var n, s = e.getPreSelection(), r = kt(e.props), o = wt(e.props), c = r && H(s, te(r)) ? r : o && U(s, Le(o)) ? o : s;
      return { open: e.props.startOpen || !1, preventFocus: !1, preSelection: (n = e.props.selectsRange ? e.props.startDate : e.props.selected) !== null && n !== void 0 ? n : c, highlightDates: pt(e.props.highlightDates), focused: !1, shouldFocusDayInline: !1, isRenderAriaLiveMessage: !1 };
    }), p(l(e), "clearPreventFocusTimeout", function() {
      e.preventFocusTimeout && clearTimeout(e.preventFocusTimeout);
    }), p(l(e), "setFocus", function() {
      e.input && e.input.focus && e.input.focus({ preventScroll: !0 });
    }), p(l(e), "setBlur", function() {
      e.input && e.input.blur && e.input.blur(), e.cancelFocusInput();
    }), p(l(e), "setOpen", function(n) {
      var s = arguments.length > 1 && arguments[1] !== void 0 && arguments[1];
      e.setState({ open: n, preSelection: n && e.state.open ? e.state.preSelection : e.calcInitialState().preSelection, lastPreSelectChange: Te }, function() {
        n || e.setState(function(r) {
          return { focused: !!s && r.focused };
        }, function() {
          !s && e.setBlur(), e.setState({ inputValue: null });
        });
      });
    }), p(l(e), "inputOk", function() {
      return We(e.state.preSelection);
    }), p(l(e), "isCalendarOpen", function() {
      return e.props.open === void 0 ? e.state.open && !e.props.disabled && !e.props.readOnly : e.props.open;
    }), p(l(e), "handleFocus", function(n) {
      e.state.preventFocus || (e.props.onFocus(n), e.props.preventOpenOnFocus || e.props.readOnly || e.setOpen(!0)), e.setState({ focused: !0 });
    }), p(l(e), "sendFocusBackToInput", function() {
      e.preventFocusTimeout && e.clearPreventFocusTimeout(), e.setState({ preventFocus: !0 }, function() {
        e.preventFocusTimeout = setTimeout(function() {
          e.setFocus(), e.setState({ preventFocus: !1 });
        });
      });
    }), p(l(e), "cancelFocusInput", function() {
      clearTimeout(e.inputFocusTimeout), e.inputFocusTimeout = null;
    }), p(l(e), "deferFocusInput", function() {
      e.cancelFocusInput(), e.inputFocusTimeout = setTimeout(function() {
        return e.setFocus();
      }, 1);
    }), p(l(e), "handleDropdownFocus", function() {
      e.cancelFocusInput();
    }), p(l(e), "handleBlur", function(n) {
      (!e.state.open || e.props.withPortal || e.props.showTimeInput) && e.props.onBlur(n), e.setState({ focused: !1 });
    }), p(l(e), "handleCalendarClickOutside", function(n) {
      e.props.inline || e.setOpen(!1), e.props.onClickOutside(n), e.props.withPortal && n.preventDefault();
    }), p(l(e), "handleChange", function() {
      for (var n = arguments.length, s = new Array(n), r = 0; r < n; r++)
        s[r] = arguments[r];
      var o = s[0];
      if (!e.props.onChangeRaw || (e.props.onChangeRaw.apply(l(e), s), typeof o.isDefaultPrevented == "function" && !o.isDefaultPrevented())) {
        e.setState({ inputValue: o.target.value, lastPreSelectChange: Sr });
        var c, d, m, f, y, g, v, D, M = (c = o.target.value, d = e.props.dateFormat, m = e.props.locale, f = e.props.strictParsing, y = e.props.minDate, g = null, v = J(m) || J(G()), D = !0, Array.isArray(d) ? (d.forEach(function(S) {
          var k = _e(c, S, /* @__PURE__ */ new Date(), { locale: v });
          f && (D = Q(k, y) && c === _(k, S, m)), Q(k, y) && D && (g = k);
        }), g) : (g = _e(c, d, /* @__PURE__ */ new Date(), { locale: v }), f ? D = Q(g) && c === _(g, d, m) : Q(g) || (d = d.match(Xt).map(function(S) {
          var k = S[0];
          return k === "p" || k === "P" ? v ? (0, Jt[k])(S, v.formatLong) : k : S;
        }).join(""), c.length > 0 && (g = _e(c, d.slice(0, c.length), /* @__PURE__ */ new Date())), Q(g) || (g = new Date(c))), Q(g) && D ? g : null));
        e.props.showTimeSelectOnly && e.props.selected && M && !C(M, e.props.selected) && (M = Gt(e.props.selected, { hours: W(M), minutes: A(M), seconds: je(M) })), !M && o.target.value || (e.props.showWeekPicker && (M = K(M, e.props.locale, e.props.calendarStartDay)), e.setSelected(M, o, !0));
      }
    }), p(l(e), "handleSelect", function(n, s, r) {
      if (e.props.shouldCloseOnSelect && !e.props.showTimeSelect && e.sendFocusBackToInput(), e.props.onChangeRaw && e.props.onChangeRaw(s), e.props.showWeekPicker && (n = K(n, e.props.locale, e.props.calendarStartDay)), e.setSelected(n, s, !1, r), e.props.showDateSelect && e.setState({ isRenderAriaLiveMessage: !0 }), !e.props.shouldCloseOnSelect || e.props.showTimeSelect)
        e.setPreSelection(n);
      else if (!e.props.inline) {
        e.props.selectsRange || e.setOpen(!1);
        var o = e.props, c = o.startDate, d = o.endDate;
        !c || d || H(n, c) || e.setOpen(!1);
      }
    }), p(l(e), "setSelected", function(n, s, r, o) {
      var c = n;
      if (e.props.showYearPicker) {
        if (c !== null && Dt(w(c), e.props))
          return;
      } else if (e.props.showMonthYearPicker) {
        if (c !== null && vt(c, e.props))
          return;
      } else if (c !== null && Se(c, e.props))
        return;
      var d = e.props, m = d.onChange, f = d.selectsRange, y = d.startDate, g = d.endDate;
      if (!z(e.props.selected, c) || e.props.allowSameDay || f)
        if (c !== null && (!e.props.selected || r && (e.props.showTimeSelect || e.props.showTimeSelectOnly || e.props.showTimeInput) || (c = Xe(c, { hour: W(e.props.selected), minute: A(e.props.selected), second: je(e.props.selected) })), e.props.inline || e.setState({ preSelection: c }), e.props.focusSelectedMonth || e.setState({ monthSelectedIn: o })), f) {
          var v = y && !g, D = y && g;
          !y && !g ? m([c, null], s) : v && (H(c, y) ? m([c, null], s) : m([y, c], s)), D && m([c, null], s);
        } else
          m(c, s);
      r || (e.props.onSelect(c, s), e.setState({ inputValue: null }));
    }), p(l(e), "setPreSelection", function(n) {
      var s = e.props.minDate !== void 0, r = e.props.maxDate !== void 0, o = !0;
      if (n) {
        e.props.showWeekPicker && (n = K(n, e.props.locale, e.props.calendarStartDay));
        var c = te(n);
        if (s && r)
          o = he(n, e.props.minDate, e.props.maxDate);
        else if (s) {
          var d = te(e.props.minDate);
          o = U(n, d) || z(c, d);
        } else if (r) {
          var m = Le(e.props.maxDate);
          o = H(n, m) || z(c, m);
        }
      }
      o && e.setState({ preSelection: n });
    }), p(l(e), "toggleCalendar", function() {
      e.setOpen(!e.state.open);
    }), p(l(e), "handleTimeChange", function(n) {
      var s = e.props.selected ? e.props.selected : e.getPreSelection(), r = e.props.selected ? n : Xe(s, { hour: W(n), minute: A(n) });
      e.setState({ preSelection: r }), e.props.onChange(r), e.props.shouldCloseOnSelect && (e.sendFocusBackToInput(), e.setOpen(!1)), e.props.showTimeInput && e.setOpen(!0), (e.props.showTimeSelectOnly || e.props.showTimeSelect) && e.setState({ isRenderAriaLiveMessage: !0 }), e.setState({ inputValue: null });
    }), p(l(e), "onInputClick", function() {
      e.props.disabled || e.props.readOnly || e.setOpen(!0), e.props.onInputClick();
    }), p(l(e), "onInputKeyDown", function(n) {
      e.props.onKeyDown(n);
      var s = n.key;
      if (e.state.open || e.props.inline || e.props.preventOpenOnFocus) {
        if (e.state.open) {
          if (s === "ArrowDown" || s === "ArrowUp") {
            n.preventDefault();
            var r = e.props.showWeekPicker && e.props.showWeekNumbers ? '.react-datepicker__week-number[tabindex="0"]' : '.react-datepicker__day[tabindex="0"]', o = e.calendar.componentNode && e.calendar.componentNode.querySelector(r);
            return void (o && o.focus({ preventScroll: !0 }));
          }
          var c = b(e.state.preSelection);
          s === "Enter" ? (n.preventDefault(), e.inputOk() && e.state.lastPreSelectChange === Te ? (e.handleSelect(c, n), !e.props.shouldCloseOnSelect && e.setPreSelection(c)) : e.setOpen(!1)) : s === "Escape" ? (n.preventDefault(), e.sendFocusBackToInput(), e.setOpen(!1)) : s === "Tab" && e.setOpen(!1), e.inputOk() || e.props.onInputError({ code: 1, msg: Ye });
        }
      } else
        s !== "ArrowDown" && s !== "ArrowUp" && s !== "Enter" || e.onInputClick();
    }), p(l(e), "onPortalKeyDown", function(n) {
      n.key === "Escape" && (n.preventDefault(), e.setState({ preventFocus: !0 }, function() {
        e.setOpen(!1), setTimeout(function() {
          e.setFocus(), e.setState({ preventFocus: !1 });
        });
      }));
    }), p(l(e), "onDayKeyDown", function(n) {
      e.props.onKeyDown(n);
      var s = n.key, r = b(e.state.preSelection);
      if (s === "Enter")
        n.preventDefault(), e.handleSelect(r, n), !e.props.shouldCloseOnSelect && e.setPreSelection(r);
      else if (s === "Escape")
        n.preventDefault(), e.setOpen(!1), e.inputOk() || e.props.onInputError({ code: 1, msg: Ye });
      else if (!e.props.disabledKeyboardNavigation) {
        var o;
        switch (s) {
          case "ArrowLeft":
            o = e.props.showWeekPicker ? He(r, 1) : Yt(r, 1);
            break;
          case "ArrowRight":
            o = e.props.showWeekPicker ? Oe(r, 1) : we(r, 1);
            break;
          case "ArrowUp":
            o = He(r, 1);
            break;
          case "ArrowDown":
            o = Oe(r, 1);
            break;
          case "PageUp":
            o = re(r, 1);
            break;
          case "PageDown":
            o = q(r, 1);
            break;
          case "Home":
            o = pe(r, 1);
            break;
          case "End":
            o = ie(r, 1);
            break;
          default:
            o = null;
        }
        if (!o)
          return void (e.props.onInputError && e.props.onInputError({ code: 1, msg: Ye }));
        if (n.preventDefault(), e.setState({ lastPreSelectChange: Te }), e.props.adjustDateOnChange && e.setSelected(o), e.setPreSelection(o), e.props.inline) {
          var c = I(r), d = I(o), m = w(r), f = w(o);
          c !== d || m !== f ? e.setState({ shouldFocusDayInline: !0 }) : e.setState({ shouldFocusDayInline: !1 });
        }
      }
    }), p(l(e), "onPopperKeyDown", function(n) {
      n.key === "Escape" && (n.preventDefault(), e.sendFocusBackToInput());
    }), p(l(e), "onClearClick", function(n) {
      n && n.preventDefault && n.preventDefault(), e.sendFocusBackToInput(), e.props.selectsRange ? e.props.onChange([null, null], n) : e.props.onChange(null, n), e.setState({ inputValue: null });
    }), p(l(e), "clear", function() {
      e.onClearClick();
    }), p(l(e), "onScroll", function(n) {
      typeof e.props.closeOnScroll == "boolean" && e.props.closeOnScroll ? n.target !== document && n.target !== document.documentElement && n.target !== document.body || e.setOpen(!1) : typeof e.props.closeOnScroll == "function" && e.props.closeOnScroll(n) && e.setOpen(!1);
    }), p(l(e), "renderCalendar", function() {
      return e.props.inline || e.isCalendarOpen() ? h.createElement(br, { ref: function(n) {
        e.calendar = n;
      }, locale: e.props.locale, calendarStartDay: e.props.calendarStartDay, chooseDayAriaLabelPrefix: e.props.chooseDayAriaLabelPrefix, disabledDayAriaLabelPrefix: e.props.disabledDayAriaLabelPrefix, weekAriaLabelPrefix: e.props.weekAriaLabelPrefix, monthAriaLabelPrefix: e.props.monthAriaLabelPrefix, adjustDateOnChange: e.props.adjustDateOnChange, setOpen: e.setOpen, shouldCloseOnSelect: e.props.shouldCloseOnSelect, dateFormat: e.props.dateFormatCalendar, useWeekdaysShort: e.props.useWeekdaysShort, formatWeekDay: e.props.formatWeekDay, dropdownMode: e.props.dropdownMode, selected: e.props.selected, preSelection: e.state.preSelection, onSelect: e.handleSelect, onWeekSelect: e.props.onWeekSelect, openToDate: e.props.openToDate, minDate: e.props.minDate, maxDate: e.props.maxDate, selectsStart: e.props.selectsStart, selectsEnd: e.props.selectsEnd, selectsRange: e.props.selectsRange, startDate: e.props.startDate, endDate: e.props.endDate, excludeDates: e.props.excludeDates, excludeDateIntervals: e.props.excludeDateIntervals, filterDate: e.props.filterDate, onClickOutside: e.handleCalendarClickOutside, formatWeekNumber: e.props.formatWeekNumber, highlightDates: e.state.highlightDates, holidays: er(e.modifyHolidays()), includeDates: e.props.includeDates, includeDateIntervals: e.props.includeDateIntervals, includeTimes: e.props.includeTimes, injectTimes: e.props.injectTimes, inline: e.props.inline, shouldFocusDayInline: e.state.shouldFocusDayInline, peekNextMonth: e.props.peekNextMonth, showMonthDropdown: e.props.showMonthDropdown, showPreviousMonths: e.props.showPreviousMonths, useShortMonthInDropdown: e.props.useShortMonthInDropdown, showMonthYearDropdown: e.props.showMonthYearDropdown, showWeekNumbers: e.props.showWeekNumbers, showYearDropdown: e.props.showYearDropdown, withPortal: e.props.withPortal, forceShowMonthNavigation: e.props.forceShowMonthNavigation, showDisabledMonthNavigation: e.props.showDisabledMonthNavigation, scrollableYearDropdown: e.props.scrollableYearDropdown, scrollableMonthYearDropdown: e.props.scrollableMonthYearDropdown, todayButton: e.props.todayButton, weekLabel: e.props.weekLabel, outsideClickIgnoreClass: dt, fixedHeight: e.props.fixedHeight, monthsShown: e.props.monthsShown, monthSelectedIn: e.state.monthSelectedIn, onDropdownFocus: e.handleDropdownFocus, onMonthChange: e.props.onMonthChange, onYearChange: e.props.onYearChange, dayClassName: e.props.dayClassName, weekDayClassName: e.props.weekDayClassName, monthClassName: e.props.monthClassName, timeClassName: e.props.timeClassName, showDateSelect: e.props.showDateSelect, showTimeSelect: e.props.showTimeSelect, showTimeSelectOnly: e.props.showTimeSelectOnly, onTimeChange: e.handleTimeChange, timeFormat: e.props.timeFormat, timeIntervals: e.props.timeIntervals, minTime: e.props.minTime, maxTime: e.props.maxTime, excludeTimes: e.props.excludeTimes, filterTime: e.props.filterTime, timeCaption: e.props.timeCaption, className: e.props.calendarClassName, container: e.props.calendarContainer, yearItemNumber: e.props.yearItemNumber, yearDropdownItemNumber: e.props.yearDropdownItemNumber, previousMonthAriaLabel: e.props.previousMonthAriaLabel, previousMonthButtonLabel: e.props.previousMonthButtonLabel, nextMonthAriaLabel: e.props.nextMonthAriaLabel, nextMonthButtonLabel: e.props.nextMonthButtonLabel, previousYearAriaLabel: e.props.previousYearAriaLabel, previousYearButtonLabel: e.props.previousYearButtonLabel, nextYearAriaLabel: e.props.nextYearAriaLabel, nextYearButtonLabel: e.props.nextYearButtonLabel, timeInputLabel: e.props.timeInputLabel, disabledKeyboardNavigation: e.props.disabledKeyboardNavigation, renderCustomHeader: e.props.renderCustomHeader, popperProps: e.props.popperProps, renderDayContents: e.props.renderDayContents, renderMonthContent: e.props.renderMonthContent, renderQuarterContent: e.props.renderQuarterContent, renderYearContent: e.props.renderYearContent, onDayMouseEnter: e.props.onDayMouseEnter, onMonthMouseLeave: e.props.onMonthMouseLeave, onYearMouseEnter: e.props.onYearMouseEnter, onYearMouseLeave: e.props.onYearMouseLeave, selectsDisabledDaysInRange: e.props.selectsDisabledDaysInRange, showTimeInput: e.props.showTimeInput, showMonthYearPicker: e.props.showMonthYearPicker, showFullMonthYearPicker: e.props.showFullMonthYearPicker, showTwoColumnMonthYearPicker: e.props.showTwoColumnMonthYearPicker, showFourColumnMonthYearPicker: e.props.showFourColumnMonthYearPicker, showYearPicker: e.props.showYearPicker, showQuarterYearPicker: e.props.showQuarterYearPicker, showWeekPicker: e.props.showWeekPicker, showPopperArrow: e.props.showPopperArrow, excludeScrollbar: e.props.excludeScrollbar, handleOnKeyDown: e.props.onKeyDown, handleOnDayKeyDown: e.onDayKeyDown, isInputFocused: e.state.focused, customTimeInput: e.props.customTimeInput, setPreSelection: e.setPreSelection }, e.props.children) : null;
    }), p(l(e), "renderAriaLiveRegion", function() {
      var n, s = e.props, r = s.dateFormat, o = s.locale, c = e.props.showTimeInput || e.props.showTimeSelect ? "PPPPp" : "PPPP";
      return n = e.props.selectsRange ? "Selected start date: ".concat(F(e.props.startDate, { dateFormat: c, locale: o }), ". ").concat(e.props.endDate ? "End date: " + F(e.props.endDate, { dateFormat: c, locale: o }) : "") : e.props.showTimeSelectOnly ? "Selected time: ".concat(F(e.props.selected, { dateFormat: r, locale: o })) : e.props.showYearPicker ? "Selected year: ".concat(F(e.props.selected, { dateFormat: "yyyy", locale: o })) : e.props.showMonthYearPicker ? "Selected month: ".concat(F(e.props.selected, { dateFormat: "MMMM yyyy", locale: o })) : e.props.showQuarterYearPicker ? "Selected quarter: ".concat(F(e.props.selected, { dateFormat: "yyyy, QQQ", locale: o })) : "Selected date: ".concat(F(e.props.selected, { dateFormat: c, locale: o })), h.createElement("span", { role: "alert", "aria-live": "polite", className: "react-datepicker__aria-live" }, n);
    }), p(l(e), "renderDateInput", function() {
      var n, s = T(e.props.className, p({}, dt, e.state.open)), r = e.props.customInput || h.createElement("input", { type: "text" }), o = e.props.customInputRef || "ref", c = typeof e.props.value == "string" ? e.props.value : typeof e.state.inputValue == "string" ? e.state.inputValue : e.props.selectsRange ? function(d, m, f) {
        if (!d)
          return "";
        var y = F(d, f), g = m ? F(m, f) : "";
        return "".concat(y, " - ").concat(g);
      }(e.props.startDate, e.props.endDate, e.props) : F(e.props.selected, e.props);
      return h.cloneElement(r, (p(p(p(p(p(p(p(p(p(p(n = {}, o, function(d) {
        e.input = d;
      }), "value", c), "onBlur", e.handleBlur), "onChange", e.handleChange), "onClick", e.onInputClick), "onFocus", e.handleFocus), "onKeyDown", e.onInputKeyDown), "id", e.props.id), "name", e.props.name), "form", e.props.form), p(p(p(p(p(p(p(p(p(p(n, "autoFocus", e.props.autoFocus), "placeholder", e.props.placeholderText), "disabled", e.props.disabled), "autoComplete", e.props.autoComplete), "className", T(r.props.className, s)), "title", e.props.title), "readOnly", e.props.readOnly), "required", e.props.required), "tabIndex", e.props.tabIndex), "aria-describedby", e.props.ariaDescribedBy), p(p(p(n, "aria-invalid", e.props.ariaInvalid), "aria-labelledby", e.props.ariaLabelledBy), "aria-required", e.props.ariaRequired)));
    }), p(l(e), "renderClearButton", function() {
      var n = e.props, s = n.isClearable, r = n.disabled, o = n.selected, c = n.startDate, d = n.endDate, m = n.clearButtonTitle, f = n.clearButtonClassName, y = f === void 0 ? "" : f, g = n.ariaLabelClose, v = g === void 0 ? "Close" : g;
      return !s || o == null && c == null && d == null ? null : h.createElement("button", { type: "button", className: T("react-datepicker__close-icon", y, { "react-datepicker__close-icon--disabled": r }), disabled: r, "aria-label": v, onClick: e.onClearClick, title: m, tabIndex: -1 });
    }), e.state = e.calcInitialState(), e.preventFocusTimeout = null, e;
  }
  return N(i, [{ key: "componentDidMount", value: function() {
    window.addEventListener("scroll", this.onScroll, !0);
  } }, { key: "componentDidUpdate", value: function(t, e) {
    var n, s;
    t.inline && (n = t.selected, s = this.props.selected, n && s ? I(n) !== I(s) || w(n) !== w(s) : n !== s) && this.setPreSelection(this.props.selected), this.state.monthSelectedIn !== void 0 && t.monthsShown !== this.props.monthsShown && this.setState({ monthSelectedIn: 0 }), t.highlightDates !== this.props.highlightDates && this.setState({ highlightDates: pt(this.props.highlightDates) }), e.focused || z(t.selected, this.props.selected) || this.setState({ inputValue: null }), e.open !== this.state.open && (e.open === !1 && this.state.open === !0 && this.props.onCalendarOpen(), e.open === !0 && this.state.open === !1 && this.props.onCalendarClose());
  } }, { key: "componentWillUnmount", value: function() {
    this.clearPreventFocusTimeout(), window.removeEventListener("scroll", this.onScroll, !0);
  } }, { key: "renderInputContainer", value: function() {
    var t = this.props, e = t.showIcon, n = t.icon, s = t.calendarIconClassname, r = t.toggleCalendarOnIconClick, o = this.state.open;
    return h.createElement("div", { className: "react-datepicker__input-container".concat(e ? " react-datepicker__view-calendar-icon" : "") }, e && h.createElement(Dr, ne({ icon: n, className: "".concat(s, " ").concat(o && "react-datepicker-ignore-onclickoutside") }, r ? { onClick: this.toggleCalendar } : null)), this.state.isRenderAriaLiveMessage && this.renderAriaLiveRegion(), this.renderDateInput(), this.renderClearButton());
  } }, { key: "render", value: function() {
    var t = this.renderCalendar();
    if (this.props.inline)
      return t;
    if (this.props.withPortal) {
      var e = this.state.open ? h.createElement(Pt, { enableTabLoop: this.props.enableTabLoop }, h.createElement("div", { className: "react-datepicker__portal", tabIndex: -1, onKeyDown: this.onPortalKeyDown }, t)) : null;
      return this.state.open && this.props.portalId && (e = h.createElement(Mt, { portalId: this.props.portalId, portalHost: this.props.portalHost }, e)), h.createElement("div", null, this.renderInputContainer(), e);
    }
    return h.createElement(wr, { className: this.props.popperClassName, wrapperClassName: this.props.wrapperClassName, hidePopper: !this.isCalendarOpen(), portalId: this.props.portalId, portalHost: this.props.portalHost, popperModifiers: this.props.popperModifiers, targetComponent: this.renderInputContainer(), popperContainer: this.props.popperContainer, popperComponent: t, popperPlacement: this.props.popperPlacement, popperProps: this.props.popperProps, popperOnKeyDown: this.onPopperKeyDown, enableTabLoop: this.props.enableTabLoop });
  } }], [{ key: "defaultProps", get: function() {
    return { allowSameDay: !1, dateFormat: "MM/dd/yyyy", dateFormatCalendar: "LLLL yyyy", onChange: function() {
    }, disabled: !1, disabledKeyboardNavigation: !1, dropdownMode: "scroll", onFocus: function() {
    }, onBlur: function() {
    }, onKeyDown: function() {
    }, onInputClick: function() {
    }, onSelect: function() {
    }, onClickOutside: function() {
    }, onMonthChange: function() {
    }, onCalendarOpen: function() {
    }, onCalendarClose: function() {
    }, preventOpenOnFocus: !1, onYearChange: function() {
    }, onInputError: function() {
    }, monthsShown: 1, readOnly: !1, withPortal: !1, selectsDisabledDaysInRange: !1, shouldCloseOnSelect: !0, showTimeSelect: !1, showTimeInput: !1, showPreviousMonths: !1, showMonthYearPicker: !1, showFullMonthYearPicker: !1, showTwoColumnMonthYearPicker: !1, showFourColumnMonthYearPicker: !1, showYearPicker: !1, showQuarterYearPicker: !1, showWeekPicker: !1, strictParsing: !1, timeIntervals: 30, timeCaption: "Time", previousMonthAriaLabel: "Previous Month", previousMonthButtonLabel: "Previous Month", nextMonthAriaLabel: "Next Month", nextMonthButtonLabel: "Next Month", previousYearAriaLabel: "Previous Year", previousYearButtonLabel: "Previous Year", nextYearAriaLabel: "Next Year", nextYearButtonLabel: "Next Year", timeInputLabel: "Time", enableTabLoop: !0, yearItemNumber: se, focusSelectedMonth: !1, showPopperArrow: !0, excludeScrollbar: !0, customTimeInput: null, calendarStartDay: void 0, toggleCalendarOnIconClick: !1 };
  } }]), i;
}(), Sr = "input", Te = "navigate";
export {
  gr as CalendarContainer,
  On as default,
  G as getDefaultLocale
};
