// react-colorful@5.6.1/dist/index.mjs
import c, { useRef as g, useMemo as R, useEffect as E, useState as V, useCallback as F, useLayoutEffect as J } from "react";
function y() {
  return (y = Object.assign || function(e) {
    for (var r = 1; r < arguments.length; r++) {
      var t = arguments[r];
      for (var n in t)
        Object.prototype.hasOwnProperty.call(t, n) && (e[n] = t[n]);
    }
    return e;
  }).apply(this, arguments);
}
function q(e, r) {
  if (e == null)
    return {};
  var t, n, o = {}, a = Object.keys(e);
  for (n = 0; n < a.length; n++)
    r.indexOf(t = a[n]) >= 0 || (o[t] = e[t]);
  return o;
}
function k(e) {
  var r = g(e), t = g(function(n) {
    r.current && r.current(n);
  });
  return r.current = e, t.current;
}
var m = function(e, r, t) {
  return r === void 0 && (r = 0), t === void 0 && (t = 1), e > t ? t : e < r ? r : e;
}, b = function(e) {
  return "touches" in e;
}, H = function(e) {
  return e && e.ownerDocument.defaultView || self;
}, K = function(e, r, t) {
  var n = e.getBoundingClientRect(), o = b(r) ? function(a, u) {
    for (var l = 0; l < a.length; l++)
      if (a[l].identifier === u)
        return a[l];
    return a[0];
  }(r.touches, t) : r;
  return { left: m((o.pageX - (n.left + H(e).pageXOffset)) / n.width), top: m((o.pageY - (n.top + H(e).pageYOffset)) / n.height) };
}, L = function(e) {
  !b(e) && e.preventDefault();
}, j = c.memo(function(e) {
  var r = e.onMove, t = e.onKey, n = q(e, ["onMove", "onKey"]), o = g(null), a = k(r), u = k(t), l = g(null), h = g(!1), i = R(function() {
    var B = function(v) {
      L(v), (b(v) ? v.touches.length > 0 : v.buttons > 0) && o.current ? a(K(o.current, v, l.current)) : x(!1);
    }, X = function() {
      return x(!1);
    };
    function x(v) {
      var f = h.current, p = H(o.current), _ = v ? p.addEventListener : p.removeEventListener;
      _(f ? "touchmove" : "mousemove", B), _(f ? "touchend" : "mouseup", X);
    }
    return [function(v) {
      var f = v.nativeEvent, p = o.current;
      if (p && (L(f), !function(Y, P) {
        return P && !b(Y);
      }(f, h.current) && p)) {
        if (b(f)) {
          h.current = !0;
          var _ = f.changedTouches || [];
          _.length && (l.current = _[0].identifier);
        }
        p.focus(), a(K(p, f, l.current)), x(!0);
      }
    }, function(v) {
      var f = v.which || v.keyCode;
      f < 37 || f > 40 || (v.preventDefault(), u({ left: f === 39 ? 0.05 : f === 37 ? -0.05 : 0, top: f === 40 ? 0.05 : f === 38 ? -0.05 : 0 }));
    }, x];
  }, [u, a]), d = i[0], M = i[1], D = i[2];
  return E(function() {
    return D;
  }, [D]), c.createElement("div", y({}, n, { onTouchStart: d, onMouseDown: d, className: "react-colorful__interactive", ref: o, onKeyDown: M, tabIndex: 0, role: "slider" }));
}), C = function(e) {
  return e.filter(Boolean).join(" ");
}, z = function(e) {
  var r = e.color, t = e.left, n = e.top, o = n === void 0 ? 0.5 : n, a = C(["react-colorful__pointer", e.className]);
  return c.createElement("div", { className: a, style: { top: 100 * o + "%", left: 100 * t + "%" } }, c.createElement("div", { className: "react-colorful__pointer-fill", style: { backgroundColor: r } }));
}, s = function(e, r, t) {
  return r === void 0 && (r = 0), t === void 0 && (t = Math.pow(10, r)), Math.round(t * e) / t;
}, Q = function(e) {
  return W(I(e));
}, I = function(e) {
  return e[0] === "#" && (e = e.substring(1)), e.length < 6 ? { r: parseInt(e[0] + e[0], 16), g: parseInt(e[1] + e[1], 16), b: parseInt(e[2] + e[2], 16), a: e.length === 4 ? s(parseInt(e[3] + e[3], 16) / 255, 2) : 1 } : { r: parseInt(e.substring(0, 2), 16), g: parseInt(e.substring(2, 4), 16), b: parseInt(e.substring(4, 6), 16), a: e.length === 8 ? s(parseInt(e.substring(6, 8), 16) / 255, 2) : 1 };
}, $ = function(e) {
  return U(G(e));
}, T = function(e) {
  var r = e.s, t = e.v, n = e.a, o = (200 - r) * t / 100;
  return { h: s(e.h), s: s(o > 0 && o < 200 ? r * t / 100 / (o <= 100 ? o : 200 - o) * 100 : 0), l: s(o / 2), a: s(n, 2) };
}, O = function(e) {
  var r = T(e);
  return "hsl(" + r.h + ", " + r.s + "%, " + r.l + "%)";
}, N = function(e) {
  var r = T(e);
  return "hsla(" + r.h + ", " + r.s + "%, " + r.l + "%, " + r.a + ")";
}, G = function(e) {
  var r = e.h, t = e.s, n = e.v, o = e.a;
  r = r / 360 * 6, t /= 100, n /= 100;
  var a = Math.floor(r), u = n * (1 - t), l = n * (1 - (r - a) * t), h = n * (1 - (1 - r + a) * t), i = a % 6;
  return { r: s(255 * [n, l, u, u, h, n][i]), g: s(255 * [h, n, n, l, u, u][i]), b: s(255 * [u, u, h, n, n, l][i]), a: s(o, 2) };
}, w = function(e) {
  var r = e.toString(16);
  return r.length < 2 ? "0" + r : r;
}, U = function(e) {
  var r = e.r, t = e.g, n = e.b, o = e.a, a = o < 1 ? w(s(255 * o)) : "";
  return "#" + w(r) + w(t) + w(n) + a;
}, W = function(e) {
  var r = e.r, t = e.g, n = e.b, o = e.a, a = Math.max(r, t, n), u = a - Math.min(r, t, n), l = u ? a === r ? (t - n) / u : a === t ? 2 + (n - r) / u : 4 + (r - t) / u : 0;
  return { h: s(60 * (l < 0 ? l + 6 : l)), s: s(a ? u / a * 100 : 0), v: s(a / 255 * 100), a: o };
}, Z = c.memo(function(e) {
  var r = e.hue, t = e.onChange, n = C(["react-colorful__hue", e.className]);
  return c.createElement("div", { className: n }, c.createElement(j, { onMove: function(o) {
    t({ h: 360 * o.left });
  }, onKey: function(o) {
    t({ h: m(r + 360 * o.left, 0, 360) });
  }, "aria-label": "Hue", "aria-valuenow": s(r), "aria-valuemax": "360", "aria-valuemin": "0" }, c.createElement(z, { className: "react-colorful__hue-pointer", left: r / 360, color: O({ h: r, s: 100, v: 100, a: 1 }) })));
}), ee = c.memo(function(e) {
  var r = e.hsva, t = e.onChange, n = { backgroundColor: O({ h: r.h, s: 100, v: 100, a: 1 }) };
  return c.createElement("div", { className: "react-colorful__saturation", style: n }, c.createElement(j, { onMove: function(o) {
    t({ s: 100 * o.left, v: 100 - 100 * o.top });
  }, onKey: function(o) {
    t({ s: m(r.s + 100 * o.left, 0, 100), v: m(r.v - 100 * o.top, 0, 100) });
  }, "aria-label": "Color", "aria-valuetext": "Saturation " + s(r.s) + "%, Brightness " + s(r.v) + "%" }, c.createElement(z, { className: "react-colorful__saturation-pointer", top: 1 - r.v / 100, left: r.s / 100, color: O(r) })));
}), A = function(e, r) {
  if (e === r)
    return !0;
  for (var t in e)
    if (e[t] !== r[t])
      return !1;
  return !0;
}, re = function(e, r) {
  return e.toLowerCase() === r.toLowerCase() || A(I(e), I(r));
};
function te(e, r, t) {
  var n = k(t), o = V(function() {
    return e.toHsva(r);
  }), a = o[0], u = o[1], l = g({ color: r, hsva: a });
  E(function() {
    if (!e.equal(r, l.current.color)) {
      var i = e.toHsva(r);
      l.current = { hsva: i, color: r }, u(i);
    }
  }, [r, e]), E(function() {
    var i;
    A(a, l.current.hsva) || e.equal(i = e.fromHsva(a), l.current.color) || (l.current = { hsva: a, color: i }, n(i));
  }, [a, e, n]);
  var h = F(function(i) {
    u(function(d) {
      return Object.assign({}, d, i);
    });
  }, []);
  return [a, h];
}
var ne = typeof window < "u" ? J : E, oe = function() {
  return typeof __webpack_nonce__ < "u" ? __webpack_nonce__ : void 0;
}, S = /* @__PURE__ */ new Map(), ae = function(e) {
  ne(function() {
    var r = e.current ? e.current.ownerDocument : document;
    if (r !== void 0 && !S.has(r)) {
      var t = r.createElement("style");
      t.innerHTML = `.react-colorful{position:relative;display:flex;flex-direction:column;width:200px;height:200px;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;cursor:default}.react-colorful__saturation{position:relative;flex-grow:1;border-color:transparent;border-bottom:12px solid #000;border-radius:8px 8px 0 0;background-image:linear-gradient(0deg,#000,transparent),linear-gradient(90deg,#fff,hsla(0,0%,100%,0))}.react-colorful__alpha-gradient,.react-colorful__pointer-fill{content:"";position:absolute;left:0;top:0;right:0;bottom:0;pointer-events:none;border-radius:inherit}.react-colorful__alpha-gradient,.react-colorful__saturation{box-shadow:inset 0 0 0 1px rgba(0,0,0,.05)}.react-colorful__alpha,.react-colorful__hue{position:relative;height:24px}.react-colorful__hue{background:linear-gradient(90deg,red 0,#ff0 17%,#0f0 33%,#0ff 50%,#00f 67%,#f0f 83%,red)}.react-colorful__last-control{border-radius:0 0 8px 8px}.react-colorful__interactive{position:absolute;left:0;top:0;right:0;bottom:0;border-radius:inherit;outline:none;touch-action:none}.react-colorful__pointer{position:absolute;z-index:1;box-sizing:border-box;width:28px;height:28px;transform:translate(-50%,-50%);background-color:#fff;border:2px solid #fff;border-radius:50%;box-shadow:0 2px 4px rgba(0,0,0,.2)}.react-colorful__interactive:focus .react-colorful__pointer{transform:translate(-50%,-50%) scale(1.1)}.react-colorful__alpha,.react-colorful__alpha-pointer{background-color:#fff;background-image:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill-opacity=".05"><path d="M8 0h8v8H8zM0 8h8v8H0z"/></svg>')}.react-colorful__saturation-pointer{z-index:3}.react-colorful__hue-pointer{z-index:2}`, S.set(r, t);
      var n = oe();
      n && t.setAttribute("nonce", n), r.head.appendChild(t);
    }
  }, []);
}, le = function(e) {
  var r = e.className, t = e.hsva, n = e.onChange, o = { backgroundImage: "linear-gradient(90deg, " + N(Object.assign({}, t, { a: 0 })) + ", " + N(Object.assign({}, t, { a: 1 })) + ")" }, a = C(["react-colorful__alpha", r]), u = s(100 * t.a);
  return c.createElement("div", { className: a }, c.createElement("div", { className: "react-colorful__alpha-gradient", style: o }), c.createElement(j, { onMove: function(l) {
    n({ a: l.left });
  }, onKey: function(l) {
    n({ a: m(t.a + l.left) });
  }, "aria-label": "Alpha", "aria-valuetext": u + "%", "aria-valuenow": u, "aria-valuemin": "0", "aria-valuemax": "100" }, c.createElement(z, { className: "react-colorful__alpha-pointer", left: t.a, color: N(t) })));
}, ue = function(e) {
  var r = e.className, t = e.colorModel, n = e.color, o = n === void 0 ? t.defaultColor : n, a = e.onChange, u = q(e, ["className", "colorModel", "color", "onChange"]), l = g(null);
  ae(l);
  var h = te(t, o, a), i = h[0], d = h[1], M = C(["react-colorful", r]);
  return c.createElement("div", y({}, u, { ref: l, className: M }), c.createElement(ee, { hsva: i, onChange: d }), c.createElement(Z, { hue: i.h, onChange: d }), c.createElement(le, { hsva: i, onChange: d, className: "react-colorful__last-control" }));
}, ce = { defaultColor: "0001", toHsva: Q, fromHsva: $, equal: re }, se = function(e) {
  return c.createElement(ue, y({}, e, { colorModel: ce }));
};
export {
  se as HexAlphaColorPicker
};
